<!DOCTYPE html>
<!-- saved from url=(0043)https://sourceforge.net/p/mageck/wiki/Home/ -->
<html lang="en" class="js flexbox canvas canvastext no-touch geolocation postmessage websqldatabase indexeddb hashchange history draganddrop websockets rgba hsla multiplebgs backgroundsize borderimage borderradius boxshadow textshadow opacity cssanimations csscolumns cssgradients cssreflections csstransforms csstransforms3d csstransitions fontface video audio localstorage sessionstorage webworkers applicationcache svg inlinesvg smil svgclippaths js canvas canvastext"><style type="text/css"></style><!--<![endif]--><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    
    <title>MAGeCK / Wiki / Home</title>
    
<meta id="project_name" name="project_name" content="mageck">
<script async="" src="./MAGeCK_Wiki_Home_files/analytics.js"></script><script src="./MAGeCK_Wiki_Home_files/modernizr.custom.90514.js"></script>
<!--[if lt IE 7 ]>
  <script src="https://a.fsdn.com/allura/nf/1438963352/_ew_/theme/sftheme/js/sftheme/dd_belatedpng.js"></script>
  <script> DD_belatedPNG.fix('img, .png_bg'); //fix any <img> or .png_bg background-images </script>
<![endif]-->
<link href="./MAGeCK_Wiki_Home_files/css" rel="stylesheet" type="text/css">
<style type="text/css">
    @font-face {
        font-family: "Pictos";
        src: url('https://a.fsdn.com/allura/nf/1438963352/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.eot');
        src: local("☺"), url('https://a.fsdn.com/allura/nf/1438963352/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.woff') format('woff'), url('https://a.fsdn.com/allura/nf/1438963352/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.ttf') format('truetype'), url('https://a.fsdn.com/allura/nf/1438963352/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.svg') format('svg');
    }
</style>
    <script type="text/javascript">
        /*jslint onevar: false, nomen: false, evil: true, css: true, plusplus: false, white: false, forin: true, on: true, immed: false */
        /*global confirm, alert, unescape, window, jQuery, $, net, COMSCORE */
    </script>
    
        <!-- ew:head_css -->

    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/hilite.css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/wiki.css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/css(1)">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/font-awesome.min.css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/tool_icon_css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/forge.css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/simplemde.min.css">
    
        <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/css(2)">
    
        
<!-- /ew:head_css -->

    
    
        <!-- ew:head_js -->

    
        <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/js"></script>
    
        
<!-- /ew:head_js -->

    

    
        <style type="text/css">
            #page-body.project---init-- #top_nav { display: none; }

#page-body.project---init-- #nav_menu_holder { display: none; margin-bottom: 0; }

#page-body.project---init-- #content_base {margin-top: 0; }
        </style>
    
    
<style>
  .hidden { display: none }
  a.notfound { color: #f00; }
</style>

    <style>.XnyGbQAQOOHNXkpbYmUqsJTkY {
        display: none
    }</style>

    
<link rel="alternate" type="application/rss+xml" title="Page RSS" href="https://sourceforge.net/p/mageck/wiki/Home/feed.rss">
<link rel="alternate" type="application/atom+xml" title="Page Atom" href="https://sourceforge.net/p/mageck/wiki/Home/feed.atom">
<link rel="alternate" type="application/rss+xml" title="Wiki RSS" href="https://sourceforge.net/p/mageck/wiki/feed.rss">
<link rel="alternate" type="application/atom+xml" title="Wiki Atom" href="https://sourceforge.net/p/mageck/wiki/feed.atom">

    
    


<script type="text/javascript">
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    function _add_tracking(prefix, tracking_id) {
        ga('create', tracking_id, {cookieDomain: 'auto', 'name': prefix});
        
        ga(prefix+'.set', 'dimension9', 'mageck');
        ga(prefix+'.set', 'dimension10', 'wiki');
        ga(prefix+'.send', 'pageview');
    }
      _add_tracking('sfnt1', 'UA-32013-6');
      _add_tracking('sfnt2', 'UA-36130941-1');
    
</script>
<style type="text/css">/**
 * Style sheet used by new LibX tooltip code
 */

/* We insert a <div> with libx-tooltip style under the body.
 * This will inherit body's style - we can't afford to inherit undesirable 
 * styles and we must redefine what we need.  OTOH, some things, e.g.
 * font-size, might be ok to be inherited to stay within the page's tone.
 */
.libx-tooltip {
    display: none;
    overflow: visible;
    padding: 5px;
    z-index: 100;
    background-color: #eee;
    color: #000;
    font-weight: normal;
    font-style: normal;
    text-align: left;
    border: 2px solid #666;
    border-radius: 5px;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
}

.libx-tooltip p {
    /* override default 1em margin to keep paragraphs inside a tooltip closer together. */
    margin: .2em;
}
</style><style type="text/css">/**
 * Style sheet used by LibX autolinking code
 */
.libx-autolink {
}

</style></head>

<body class="wiki-Home" id="forge">
<h2 class="hidden">
    <span style="color:red">Error:</span> CSS did not load.<br>
    This may happen on the first request due to CSS mimetype issues.
    Try clearing your browser cache and refreshing.
    <hr>
</h2>

    
        <!-- ew:body_top_js -->

    
        
<!-- /ew:body_top_js -->

    


<header id="site-header">
    <div class="wrapper">
        <a href="https://sourceforge.net/" class="logo">
            <span>SourceForge</span>
        </a>
        
        <form method="get" action="https://sourceforge.net/directory/">
            <input type="text" id="words" name="q" placeholder="Search" tabindex="1">
        </form>
        
        <!--Switch to {language}-->
        <nav id="nav-site">
            <a href="https://sourceforge.net/directory/" title="Browse our software.">Browse</a>
            <a href="https://sourceforge.net/directory/enterprise" title="Browse our Enterprise software.">Enterprise</a>
            <a href="https://sourceforge.net/blog/" title="Read the latest news from the SF HQ.">Blog</a>
            <a href="https://sourceforge.net/jobs?source=header" title="Search 80k+ tech jobs.">Jobs</a>
            <a href="https://deals.sourceforge.net/?utm_source=sourceforge&utm_medium=navbar&utm_campaign=homepage" title="Discover and Save on the Best Gear, Gadgets, and Software" class="featured-link" target="_blank">Deals</a>
            <a href="https://sourceforge.net/support" title="Contact us for help and feedback.">Help</a>
        </nav>
        <nav id="nav-account">
            
            <div id="account-tooltip" class="tooltip-container">
                <a href="https://sourceforge.net/users/davidliwei">
                    
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5" alt="Wei Li" title="Wei Li" class="emboss x20">
  
                    Me ▾
                </a>
                <div class="tooltip account tooltipstered">
                    <div class="diamond"></div>
                    <ul>
                        <li>
                            
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5(1)" alt="Wei Li" title="Wei Li" class="emboss x20">
  
                            <a href="https://sourceforge.net/users/davidliwei">Profile</a>
                        </li>
                        <li>
                            <b data-icon="U" class="ico ico-user" title="Account"></b>
                            <a href="https://sourceforge.net/auth/preferences/">Account Settings</a>
                        </li>
                        <li>
                            <b data-icon="Q" class="ico ico-power" title="Log Out"></b>
                            <a href="https://sourceforge.net/auth/logout">Log Out</a>
                        </li>
                    </ul>
                </div>
            </div>
             <!-- Change this when update notifications are available in Allura -->
            <div id="updater-tooltip" class="tooltip-container">
                
  
<a href="https://sourceforge.net/user/updates" title="Manage Update Notifications" class=" updates ">6</a>
<div class="tooltip updater tooltipstered">
    <h6>Available Updates</h6>
    <div class="diamond"></div>
    <ul>
        <li>The following software titles have updates:</li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/mageck/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="MAGeCK" href="https://sourceforge.net/projects/mageck/?source=updater">MAGeCK</a>
            <a href="https://sourceforge.net/projects/mageck/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/scipy/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/icon"></a>
            <a title="SciPy: Scientific Library for Python" href="https://sourceforge.net/projects/scipy/?source=updater">SciPy: Scientific ...</a>
            <a href="https://sourceforge.net/projects/scipy/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/matplotlib/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/icon(1)"></a>
            <a title="matplotlib" href="https://sourceforge.net/projects/matplotlib/?source=updater">matplotlib</a>
            <a href="https://sourceforge.net/projects/matplotlib/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/keepass/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/icon(2)"></a>
            <a title="KeePass" href="https://sourceforge.net/projects/keepass/?source=updater">KeePass</a>
            <a href="https://sourceforge.net/projects/keepass/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/seqgene/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="seqgene" href="https://sourceforge.net/projects/seqgene/?source=updater">seqgene</a>
            <a href="https://sourceforge.net/projects/seqgene/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/jtreeview/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="Java Treeview" href="https://sourceforge.net/projects/jtreeview/?source=updater">Java Treeview</a>
            <a href="https://sourceforge.net/projects/jtreeview/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li><a href="https://sourceforge.net/user/updates">Manage Subscriptions</a></li>
    </ul>
</div>
            </div>
            
        </nav>
        
    </div>
</header>
<header id="site-sec-header">
    <div class="wrapper">
        <nav id="nav-hubs">
            <h4>Solution Centers</h4>
            <a href="http://goparallel.sourceforge.net/">Go Parallel</a>
            <a href="http://apmsolutions.sourceforge.net/">Performance Central</a>
        </nav>
        <nav id="nav-collateral">
            <a href="http://library.slashdotmedia.com/?source=sfnet_header">Resources</a>
            
        </nav>
    </div>
</header>

    
    
    

<section id="page-body" class=" neighborhood-Projects project-mageck mountpoint-wiki">
    <div id="nav_menu_holder">
        
            



    
    
    
    
    <nav id="breadcrumbs">
        <ul>
            <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/">Home</a></li>
            <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/directory">Browse</a></li>
            
            
                
            
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/p/">Projects</a></li>
                
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/p/mageck/">MAGeCK</a></li>
                
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb">Wiki</li>
                
            
        </ul>
    </nav>
    
    
    
  
    
    <h1 class="project_title">
        <a href="https://sourceforge.net/p/mageck/" class="project_link">MAGeCK</a>
    </h1>
    
    
    
    <h2 class="project_summary">
        Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout 
    </h2>
    
    <div class="brought-by">
        Brought to you by:
        
        
            
                <a href="https://sourceforge.net/u/davidliwei/">davidliwei</a>
            </div>
    

        
    </div>
    <div id="top_nav" class="">
        
            
<ul class="dropdown">
  
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/" class="ui-icon-tool-summary-32">
            Summary
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/files/" class="ui-icon-tool-files-32">
            Files
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/reviews" class="ui-icon-tool-reviews-32">
            Reviews
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/support" class="ui-icon-tool-support-32">
            Support
        </a>
        
        
    </li>
	
    <li class="selected">
        <a href="https://sourceforge.net/p/mageck/wiki/" class="ui-icon-tool-wiki-32">
            Wiki
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/discussion/" class="ui-icon-tool-discussion-32">
            Discussion
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/tickets/" class="ui-icon-tool-tickets-32">
            Tickets
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/admin/" class="ui-icon-tool-admin-32">
            Admin
        </a>
        
        
    </li>
	
</ul>

        
    </div>
    <div id="content_base">
        
            
                
                    


<div id="sidebar">
  
    <form id="search" method="GET" action="https://sourceforge.net/p/mageck/wiki/search/">
      <input name="q" type="text" title="Search Wiki" placeholder="Search Wiki" tabindex="2">
    </form>
  
    
    
      
        
    
      <ul class="sidebarmenu">
      
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/" class="add_wiki_page">
       <b data-icon="+" class="ico ico-plus"></b>
      <span>Create Page</span>
       </a>
  </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/" class="wiki_home">
      
      <span>Wiki Home</span>
       </a>
  </li>
  
      
        
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/browse_pages/">
      
      <span>Browse Pages</span>
       </a>
  </li>
  
      
        
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/browse_tags/">
      
      <span>Browse Labels</span>
       </a>
  </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/#toggle-subscribe">
       <b data-icon="M" class="ico ico-mail"></b>
      <span>Subscribe to wiki</span>
       </a>
  </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
  <li>
      <a href="https://sourceforge.net/p/mageck/wiki/markdown_syntax/">
      
      <span>Formatting Help</span>
       </a>
  </li>
  
      
    
    
      </ul>
      
    
    
</div>
                
                
            
            
                
            
            <div class="grid-20 pad">
                <h2 class="dark title">Home
                    <!-- actions -->
                    <small>
                        
  
    
      <a href="https://sourceforge.net/p/mageck/wiki/Home/edit" title="Edit"><b data-icon="p" class="ico ico-pencil" title="Edit"></b>&nbsp;Edit</a>
    
    <a href="https://sourceforge.net/p/mageck/wiki/Home/history" title="History"><b data-icon="N" class="ico ico-history" title="History"></b></a>
  
  
    






  <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/js(1)"></script>





<div id="subscription-form" style="display: inline;"><a class="tooltipstered" href="https://sourceforge.net/p/mageck/wiki/Home/#" data-reactid=".0"><b data-icon="M" class="ico ico-mail" title="Subscribe to this page" data-reactid=".0.0"></b></a></div>

<script>
  document.SUBSCRIPTION_OPTIONS = {
    "thing": "page",
    "subscribed": false,
    "subscribed_to_tool": false,
    "url": "subscribe",
    "icon": {
      "char": "M",
      "css": "ico-mail"
    }
  };
</script>
  
	<a href="https://sourceforge.net/p/mageck/wiki/Home/feed" title="RSS"><b data-icon="f" class="ico ico-feed" title="Feed"></b></a>
  <a href="https://sourceforge.net/p/mageck/wiki/search" title="Search"><b data-icon="s" class="ico ico-search" title="Search"></b></a>

                    </small>
                    <!-- /actions -->
                </h2>
                
  
<div class="editbox">
  
  <div class="grid-9 gravatar sm">
    <label class="simple">Authors: </label>
    
      
        
            <a href="https://sourceforge.net/u/davidliwei/">
                
                    
    
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5(1)" alt="Wei Li" title="Wei Li" class="emboss x16">
  
  
                
            </a>
        
    
    
  </div>
   
</div>


                <div>
                    
                    
  
<div class="markdown_content"><p><strong>Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout (MAGeCK)</strong> is a computational tool to identify important genes from the recent genome-scale CRISPR-Cas9 knockout screens (or GeCKO) technology. MAGeCK is developed and maintained by Wei Li and Han Xu from <a class="" href="http://liulab.dfci.harvard.edu/" rel="nofollow">Dr. Xiaole Shirley Liu's lab</a> at Department of Biostatistics and Computational Biology, Dana-Farber Cancer Institute and Harvard School of Public Health. The MAGeCK algorithm is described in the following paper:</p>
<p><a class="" href="http://genomebiology.com/2014/15/12/554/abstract" rel="nofollow">Li, et al. MAGeCK enables robust identification of essential genes from genome-scale CRISPR/Cas9 knockout screens. Genome Biology 15:554 (2014)</a></p>
<p>MAGeCK is a free, open source software under the BSD license.</p>
<p>Besides MAGeCK, we also present <a class="" href="https://bitbucket.org/liulab/mageck-vispr" rel="nofollow">MAGeCK-VISPR</a>, a comprehensive quality control, analysis and visualization workflow for CRISPR/Cas9 screens. MAGeCK-VISPR allows you to check the qualities of the screening sample, perform multiple condition comparisons (instead of only two condition comparisons in MAGeCK), and explore the results interactively.</p>
<p><strong>Any questions about MAGeCK or MAGeCK-VISPR?</strong> Check the FAQ below or join our <a class="" href="https://groups.google.com/d/forum/mageck" rel="nofollow">MAGeCK Google group</a>.</p>
<p>This documentation includes the following items:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/install/">Installation</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/history/">Version history</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/demo/">Tutorial</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/usage/">Usage</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/visualization/">Visualization</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/input/">File formats (input)</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/output/">File formats (output)</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/libraries/">Commonly used libraries</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/QA/">Frequently Asked Questions</a></li>
</ul>
<p>Take a look at the following videos to install and use MAGeCK:</p>
<ul>
<li>Installation</li>
</ul>
<div class="grid-20"><iframe height="270" src="./MAGeCK_Wiki_Home_files/0FLWSp1tdTI.html" width="480"></iframe>

<ul>
<li>Comparison between samples</li>
</ul>
<div class="grid-20"><iframe height="270" src="./MAGeCK_Wiki_Home_files/iIyP2w44ric.html" width="480"></iframe>

<div>
  
    <div class="markdown_content"><h1 id="installation">Installation</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation">Installation</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#download-and-get-the-unzip-password">Download and get the unzip password</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#system-requirement">System requirement</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#optional-dependencies">Optional dependencies</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#what-if-i-cannot-install-these-r-packages-in-my-machine">What if I cannot install these R packages in my machine</a></li>
</ul>
</li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation_1">Installation</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#wzxhzdk9-manual-installation-wzxhzdk10"><s> Manual installation </s></a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#setting-up-the-environment-variables">Setting up the environment variables</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#experimental-version">Experimental version</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="download-and-get-the-unzip-password">Download and get the unzip password</h2>
<p>The latest version of MAGeCK (0.5.1) can be downloaded here:<br>
</p><p><span class="download-button-533ec35981b24b2aced2594a" style="margin-bottom: 1em; display: block;">

    <style type="text/css" scoped="">.sfdl { width:165px;height:39px;padding:0;position:relative;font:bold 12px/1.17 sans-serif !important; border:1px solid #aaa;display:inline;float:left;text-decoration:none;-moz-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-webkit-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-o-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-moz-border-radius:4px;-webkit-border-radius:4px;-o-border-radius:4px;-ms-border-radius:4px;-khtml-border-radius:4px;border-radius:4px;background:no-repeat #e5e5e5 8px center url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC);background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), -webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #fff), color-stop(100%, #ccc));background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), -moz-linear-gradient(top, #fff 0%, #ccc 100%);background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), linear-gradient(top, #fff 0%, #ccc 100%);background-position:8px center, 0 0;margin:0 10px 0px 0;overflow:hidden;color:#555 !important;}.sfdl:hover { cursor:pointer;text-decoration:none;background:8px center no-repeat #eee url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC);}.sfdl-lite { background:0 center no-repeat #337733 none;color:white !important;height:auto;padding:0.5em 1em;text-align:center;-moz-border-radius:4px;-webkit-border-radius:4px;-o-border-radius:4px;-ms-border-radius:4px;-khtml-border-radius:4px;border-radius:4px;background-image:-webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #339933), color-stop(100%, #337733));background-image:-moz-linear-gradient(top, #339933 0%, #337733 100%);background-image:linear-gradient(top, #339933 0%, #337733 100%);}.sfdl-lite:hover { background:0 center no-repeat #339933;}.sfdl span { border-radius:3px;-moz-border-radius:3px;-webkit-border-radius:3px;display:block;margin:3px;margin-left:35px;padding:4px;text-align:left;color:#FFF;background:#363;background-image:-moz-linear-gradient(center top, #393 0%, #373 100%);background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0, #393),color-stop(1, #373));background-image:linear-gradient(top, #393 0%, #373 100%);overflow:hidden;height:25px;}.sfdl b { display:block;line-height:12.5px;}.sfdl small { display:block;line-height:12.5px;color:#ADC2AD; font-weight:normal;white-space:nowrap;overflow:hidden;font-size:77%;}.info-circle{ right:10px;bottom:16px;}.direct-dl{ font-size:10px;text-align:right;display:inline-block;width:100%;}</style>


<a href="https://sourceforge.net/projects/mageck/files/latest/download" title="Download /libraries/tim.library.txt.zip from SourceForge  - 1.4 MB" class="sfdl">
    <span>
        <b>Download</b>
        <small>tim.library.txt.zip</small>
        
    </span>
</a>
</span></p><p></p>
<hr>
<p><strong>The zip file is encrypted</strong>, but you can get the password easily by one of the following options:</p>
<ul>
<li>Join our <a class="" href="https://groups.google.com/d/forum/mageck" rel="nofollow">MAGeCK Google group</a> and the password is shown on the top of the forum; or</li>
<li>Send an email to <a class="" href="mailto:mageck.help@gmail.com">mageck.help@gmail.com</a> with the subject "password", and you will get an automatic reply with password within 1 minute.</li>
</ul>
<p>You need to go to the Terminal to unzip and install the software. See the instructions for Installation below.</p>
<hr>
<h2 id="system-requirement">System requirement</h2>
<p>MAGeCK can be run on either Mac or Linux system. Since MAGeCK is written in Python and C, Python 2.7 (&gt;2.7) and a C compiler is needed. </p>
<h3 id="optional-dependencies">Optional dependencies</h3>
<p>Two optional softwares include <a class="" href="http://cran.r-project.org/" rel="nofollow">R</a> and <a class="" href="http://www.latex-project.org/" rel="nofollow">pdflatex</a>. MAGeCK relies on both softwares to generating PDF reports if the <strong>--pdf-report</strong> option is used. If it is not possible to install them, you can also generate PDF reports by copying some MAGeCK output files to another computers with R and pdflatex are properly installed. See <a class="" href="https://sourceforge.net/p/mageck/wiki/QA/">Q and A</a> for more information. </p>
<p>If you use the <strong>--pdf-report</strong> option, <a class="" href="http://cran.r-project.org/web/packages/xtable/index.html" rel="nofollow">xtable</a> is required, and <a class="" href="http://cran.r-project.org/web/packages/gplots/index.html" rel="nofollow">gplots</a> is suggested to install in R. Use <em>install.packages("xtable")</em> and <em>install.packages("gplots")</em> in R to install them.<br>
</p>
<p>You won't get any error messages if you don't have gplots, but you will get a more beautiful clustering figure in the pdf report of the <em>count</em> command.</p>
<p>MAGeCK also suggests users installing <a class="" href="http://www.numpy.org/" rel="nofollow">numpy</a> to calculate the negative binomial p value. However, in cases where <em>numpy</em> is not found, MAGeCK will use the normal p value instead. There may be slight differences between both methods.</p>
<h3 id="what-if-i-cannot-install-these-r-packages-in-my-machine">What if I cannot install these R packages in my machine</h3>
<p>You can run MAGeCK without --pdf-report option, and copy some files to another machine with these R packages to generate pdf report. See <a class="" href="https://sourceforge.net/p/mageck/wiki/QA/#the-test-or-count-command-is-successful-but-i-have-some-problems-producing-the-pdf-file-how-can-i-generate-the-pdf-file">Q and A</a> for more details. </p>
<h2 id="installation_1">Installation</h2>
<p>Since version 0.3, MAGeCK uses standard Python installation procedures (distutils) for compiling and installation of the software. </p>
<p>The installation procedure is extremely easy. First, download the source code, unzip it (assuming the current version is 0.4.1), type the unzip password, and go into the directory:</p>
<div class="codehilite"><pre>unzip mageck-0.4.1.zip
cd mageck-0.4
</pre></div>


<p>After that, invoke python setup.py:</p>
<div class="codehilite"><pre>python setup.py install
</pre></div>


<p>And it is done! <b> If you want MAGeCK to be installed on your own directory </b>, use the following command instead:</p>
<div class="codehilite"><pre>python setup.py install --user
</pre></div>


<p>This is the easiest way to install mageck. An alternative approach is (you may have one additional step to set up the environment variables; see below)</p>
<div class="codehilite"><pre><span class="x">python setup.py install --prefix=</span><span class="p">$</span><span class="nv">HOME</span><span class="x"></span>
</pre></div>


<p>where $HOME is the root directory you want to install (usually the user home).</p>
<h2 id="wzxhzdk9-manual-installation-wzxhzdk10"><s> Manual installation </s></h2>
<p><b> The manual installation is deprecated since version 0.3. Please refer to the installation instructions above. </b></p>
<p>After downloading the source code, follow the instructions below for manual installation.</p>
<h2 id="setting-up-the-environment-variables">Setting up the environment variables</h2>
<p>In most systems you don't need to set up the environment variables. Just type "mageck" in the command line to see if the mageck program works.</p>
<p>For those who installed MAGeCK into custom directories, there may be additional steps, including changing the environment variable. First you need to add the path of the mageck program to your <strong>PATH</strong> variable. For example, if you used the <strong>--prefix=$HOME</strong> option during installation, then set up the <strong>PATH</strong> variable by typing:</p>
<div class="codehilite"><pre><span class="x">export PATH=</span><span class="p">$</span><span class="nv">PATH</span><span class="x">:</span><span class="p">$</span><span class="nv">HOME</span><span class="x">/bin</span>
</pre></div>


<p>You also need to add the path of the MAGeCK module to the <strong>PYTHONPATH</strong> variable. Again, his variable should be set as </p>
<div class="codehilite"><pre><span class="x">export PYTHONPATH=</span><span class="p">$</span><span class="nv">HOME</span><span class="x">/lib/python2.7/site-packages:</span><span class="p">$</span><span class="nv">PYTHONPATH</span><span class="x"></span>
</pre></div>


<p>To save the path configuration (so you don't have to type it every time), place the above command in your <em>~/.bashrc</em> (for Linux) or <em>~/.bash_profile</em> (for Mac).</p>
<h2 id="experimental-version">Experimental version</h2>
<p>The experimental version of MAGeCK is available at <a class="" href="https://bitbucket.org/liulab/mageck/" rel="nofollow">bitbucket</a>. <strong>Note that the source codes on BitBucket are experimental and are not fully tested, and it may not be stable or function well. It is strongly recommended to use the MAGeCK software downloaded from sourceforge.</strong><br>
</p>
<p>Return to <a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p>
<hr></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="version-history">Version history</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#version-history">Version history</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#051">0.5.1</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#05">0.5</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#044">0.4.4</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#043">0.4.3</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#042">0.4.2</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#041">0.4.1</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#04">0.4</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#03">0.3</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#02">0.2</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#01">0.1</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="051">0.5.1</h2>
<p>2015.06.23 Version 0.5.1</p>
<ul>
<li>Add one real dataset workflow in the documentation</li>
<li>MAGeCK can now run on python 3 (still experimental)</li>
<li>FDR calculation method can be specified on both sgRNA and gene level</li>
<li>The column header of the output has changed to a more human readable form</li>
<li>Add several more statistics to read count summary</li>
</ul>
<h2 id="05">0.5</h2>
<p>2015.04.26 Version 0.5</p>
<ul>
<li>Add multiple visualization functions (still experimental).</li>
<li>Fix a bug in negative binomial calculation (thanks to Ido Tamir).</li>
</ul>
<h2 id="044">0.4.4</h2>
<p>2015.03.19 Version 0.4.4</p>
<ul>
<li>Improve the running speed of the RRA program.</li>
<li>For gene testing, MAGeCK now accepts multiple -t and -c pairs, allowing generating one summary table containing results of multiple comparisons. </li>
<li>Modify the format of gene_summary.txt; the duplicated "item" column is now removed for positive selection results. Also, two more columns are added to better help users identify true hits: "lo": the RRA lo values; "goodsgrna": the number of sgRNAs in this gene whose ranking is higher than the alpha threshold. </li>
<li>MAGeCK now allows users specifying a set of control sgRNAs to generate null distributions.</li>
<li>Fix two bugs in calculating the median factor during normalization (thanks to Bastiaan Evers).</li>
<li>Add the "-v/--version" command.</li>
</ul>
<h2 id="043">0.4.3</h2>
<p>2015.02.12 Version 0.4.3</p>
<ul>
<li>Fix a bug where the program exits unexpectedly for certain samples with many 0 read counts.</li>
<li>Fix a bug of pathway analysis where the RRA program stops early for certain gene belonging to too many pathways.</li>
</ul>
<h2 id="042">0.4.2</h2>
<p>2015.02.04 Version 0.4.2</p>
<ul>
<li>Create youtube tutorial videos for installation and sample comparisons.</li>
<li>Improve the median normalization method to handle cases with many zero-count sgRNAs.</li>
<li>The median normalized read count are provided in the count command.</li>
<li>Modify the count command line options to accept combining reads from technical replicates.</li>
<li>Provide simple statistics for processing fastq files.</li>
<li>Provide library file for Synergistic Activation Mediators (SAM), a CRISPR activation protocol developed in Feng Zhang laboratory (http://www.addgene.org/crispr/libraries/sam/).</li>
</ul>
<h2 id="041">0.4.1</h2>
<p>2014.12.01 Version 0.4.1</p>
<ul>
<li>Increase the default alpha cutoff from 0.05 to 0.25.</li>
<li>Provide some of the commonly used library files for the convenience of users.</li>
</ul>
<h2 id="04">0.4</h2>
<p>2014.11.13 Version 0.4</p>
<ul>
<li>Added the BSD license information.</li>
<li>Improved the logging system.</li>
<li>The control_id and treatment_id options now can be specified using sample strings.</li>
<li>Merge positive selection and negative selection genes and pathways into one file.</li>
<li>Add the --keep-tmp option to control intermediate files after running.</li>
<li>Fixed one bug in FDR calculation.</li>
</ul>
<h2 id="03">0.3</h2>
<p>2014.07.01 Version 0.3</p>
<ul>
<li>The installation method is changed so users can now more easily install the software.</li>
<li>Added a new feature to detect enriched pathways (pathway command)</li>
<li>Changed the input format of the program:<ul>
<li>The second column of the count table (generated by the count subcommand and used by the test subcommand) is now the gene name.</li>
<li>For the count subcommand, the sgRNA information is provided with the library file.</li>
</ul>
</li>
</ul>
<h2 id="02">0.2</h2>
<p>2014.04.17 Version 0.2</p>
<ul>
<li>Updated the demo and wiki page</li>
</ul>
<h2 id="01">0.1</h2>
<p>2014.04.04 Version 0.1</p>
<ul>
<li>The source code released.</li>
</ul></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="tutorial">Tutorial</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#tutorial">Tutorial</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-first-demo-starting-from-read-count-tables">The first demo: starting from read count tables</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-second-demo-starting-from-raw-fastq-files">The second demo: starting from raw fastq files</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-third-demo-going-through-a-public-crisprcas9-screening-dataset">The third demo: going through a public CRISPR/Cas9 screening dataset</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-1-download-the-fastq-file">Step 1: download the fastq file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-2-prepare-the-library-file">Step 2: prepare the library file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-3-determine-the-trimming-length-and-sgrna-length">Step 3: determine the trimming length and sgRNA length</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-4-run-the-mageck-count-command">Step 4: run the MAGeCK count command</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-5-compare-samples-using-mageck-test-subcommand">Step 5: compare samples using MAGeCK test subcommand</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#final-remarks">Final remarks</a></li>
</ul>
</li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-fourth-demo-using-mageck-mle-module">The fourth demo: using MAGeCK mle module</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-1-download-the-count-table">Step 1: download the count table</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-2-prepare-the-design-matrix-file">Step 2: prepare the design matrix file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-3-run-the-module">Step 3: run the module</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#step-4-optional-including-the-sgrna-efficiency-into-mle-calculation">Step 4 (optional): including the sgRNA efficiency into mle calculation</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<p>Running MAGeCK is extremely easy and convenient. The <em>demo</em> folder contains two mini examples to go through all steps in MAGeCK. Simply execute the <em>sh</em> script in the command line in each example to run the demos. To see how you can enable visualization functions of MAGeCK in both demos, see the <a class="" href="https://sourceforge.net/p/mageck/wiki/visualization/">visualization</a> manual.</p>
<p>Also check out the following videos in YouTube to learn how to install and run MAGeCK:</p>
<p><a class="" href="https://www.youtube.com/watch?v=0FLWSp1tdTI" rel="nofollow">Tutorial 1: Installation</a></p>
<p><a class="" href="https://www.youtube.com/watch?v=iIyP2w44ric" rel="nofollow">Tutorial 2: Comparison between samples</a></p>
<h2 id="the-first-demo-starting-from-read-count-tables">The first demo: starting from read count tables</h2>
<p>There is only one command line in the demo:</p>
<div class="codehilite"><pre>mageck <span class="nb">test</span> -k sample.txt -t HL60.final,KBM7.final -c HL60.initial,KBM7.initial  -n demo
</pre></div>


<p>The parameters are explained as follows.</p>
<table>
<thead>
<tr>
<th>Parameters</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td>mageck</td>
<td>The main portal of the MAGeCK program</td>
</tr>
<tr>
<td>test</td>
<td>A sub-command to ask MAGeCK to perform sgRNA and gene ranking based on provided read count tables</td>
</tr>
<tr>
<td>-k sgrna_count.txt</td>
<td>The provided read count table file. The format of the file is specified <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-read-count-file">here</a>.</td>
</tr>
<tr>
<td>-t HL60.final,KBM7.final</td>
<td>The treatment samples are defined as HL60.final,KBM7.final (or the 2nd and 3rd sample, starting from 0) in sgrna_count.txt. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-c HL60.initial,KBM7.initial</td>
<td>The control samples are defined as HL60.initial,KBM7.initial (or the 0th and 1st sample, starting from 0) in sgrna_count.txt. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-n demo</td>
<td>The prefix of the output files is demo, so you will expect the output files are: demo.sgrna_summary.txt, demo.gene_summary.txt, etc.</td>
</tr>
</tbody>
</table>
<p>An explanation of the output files can be found in the <a class="alink" href="https://sourceforge.net/p/mageck/wiki/output/">[output]</a> page. For all available parameters, see the <a class="alink" href="https://sourceforge.net/p/mageck/wiki/usage/">[usage]</a> page.</p>
<p>You can also specify the treatment and control samples using sample index. For example,</p>
<div class="codehilite"><pre>mageck <span class="nb">test</span> -k sgrna_count.txt -t 2,3 -c 0,1 -n demo
</pre></div>


<h2 id="the-second-demo-starting-from-raw-fastq-files">The second demo: starting from raw fastq files</h2>
<p>This demo shows an mini example of how to go through the whole pipeline from raw fastq files. In this example, we have fastq files from two conditions, and we would like to compare which gene and sgRNA is significant between conditions. The command line used in the <em>runmageck.sh</em> script is:</p>
<div class="codehilite"><pre>mageck run  -l library.txt -n demo --sample-label L1,CTRL -t CTRL -c L1 --fastq test1.fastq test2.fastq
</pre></div>


<p>The parameters are explained as follows.</p>
<table>
<thead>
<tr>
<th>Parameters</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td>mageck</td>
<td>The main portal of the MAGeCK program</td>
</tr>
<tr>
<td>run</td>
<td>A sub-command to ask MAGeCK to go through sgRNA counting, sgRNA and gene ranking from fastq files.</td>
</tr>
<tr>
<td>-l library.txt</td>
<td>The provided sgRNA information, including the sgRNA id, the sequence, and the gene it is targeting. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-n demo</td>
<td>The prefix of the output files is demo, so you will expect the output files are: demo.summary.txt, demo.gene.high.txt, etc.</td>
</tr>
<tr>
<td>--sample-label L1,CTRL</td>
<td>The labels of the two samples are L1 (test1.fastq) and CTRL (test2.fastq).</td>
</tr>
<tr>
<td>-t CTRL</td>
<td>The treatment samples are defined as CTRL. In another way, it is the sample in test2.fastq. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-c L1</td>
<td>The control samples are defined as L1. It is the sample in test1.fastq. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>--fastq test1.fastq test2.fastq</td>
<td>The provided fastq file, separated by space. (Technical replicates of the same sample can also indicated using comma as a separator; for example, "sample1_replicate1.fastq,sample1_replicate2.fastq")</td>
</tr>
</tbody>
</table>
<p>You can also use</p>
<div class="codehilite"><pre>mageck run -l library.txt -n demo --sample-label L1,CTRL -t <span class="m">1</span> -c <span class="m">0</span> --fastq test1.fastq test2.fastq
</pre></div>


<h2 id="the-third-demo-going-through-a-public-crisprcas9-screening-dataset">The third demo: going through a public CRISPR/Cas9 screening dataset</h2>
<p>After the first two demos, you have a basic sense of how MAGeCK works. In this demo, let us go through a real dataset which is more complicated, and see how to handle some practical problems, like the trimming of the 5' end. </p>
<p>The dataset we use comes from the following paper: <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24535568" rel="nofollow">Genome-wide recessive genetic screening in mammalian cells with a lentiviral CRISPR-guide RNA library</a>. In this paper, the authors did the CRISPR/Cas9 screening on mouse ESC cells, and identify genes that are essential in mouse ESC cells.</p>
<h3 id="step-1-download-the-fastq-file">Step 1: download the fastq file</h3>
<p>The fastq files of screens are public available on <a class="" href="http://www.ebi.ac.uk/ena/data/view/ERP003292" rel="nofollow">ENA</a> archive. There are different replicates for one condition, but for simplicity, let us only download the following two fastq files and use them to test MAGeCK functions.</p>
<table>
<thead>
<tr>
<th>Accession</th>
<th>Sample</th>
<th>Download Link</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="" href="http://www.ebi.ac.uk/ena/data/view/ERR376998" rel="nofollow">ERR376998</a></td>
<td>one replicate of plasmid</td>
<td><a class="" href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR376/ERR376998/ERR376998.fastq.gz" rel="nofollow">ERR376998</a></td>
</tr>
<tr>
<td><a class="" href="http://www.ebi.ac.uk/ena/data/view/ERR376999" rel="nofollow">ERR376999</a></td>
<td>one replicate of ESC</td>
<td><a class="" href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR376/ERR376999/ERR376999.fastq.gz" rel="nofollow">ERR376999</a></td>
</tr>
</tbody>
</table>
<p>You can download these files, double click to unzip them (or use gunzip in the terminal), and place them into one separate folder:</p>
<div class="codehilite"><pre>gunzip ERR376998.fastq.gz
gunzip ERR376998.fastq.gz
</pre></div>


<h3 id="step-2-prepare-the-library-file">Step 2: prepare the library file</h3>
<p>The next step is to prepare the library file so MAGeCK will know which sgRNA targets which gene. If you are using one of the standard GeCKO libraries, you can just download the files <a class="" href="http://sourceforge.net/p/mageck/wiki/libraries/">from MAGeCK sourceforge</a>. For non-standard libraries, you need to prepare the library file according to the <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">library file format</a>. </p>
<p>In this demo, you can generate the library file using Supplementary Data 2 (or Supplementary Table S7) from the paper, or download it directly from <a class="" href="http://sourceforge.net/p/mageck/wiki/libraries/">our collection of libraries</a> (the file name is "yusa_library.csv.zip). Double click to unzip it (or use "unzip" in the terminal).</p>
<h3 id="step-3-determine-the-trimming-length-and-sgrna-length">Step 3: determine the trimming length and sgRNA length</h3>
<p>In many cases, your sequencing primer is not exactly in front of the first base of guide RNA. This is indeed the case in this demo, where the the first few bases in the fastq file are identical. Make sure you know exactly how many bases to trim before running MAGeCK. You can talk to experimental people, or get this information by taking a look at the first few lines of the fastq files.</p>
<p>Here are the first few lines of ERR376998.fastq (only sequences are shown):</p>
<p><strong>CTTGTGGAAAGGACGAAACACCG</strong>GTGAAGGTGCCGTTGTGTAGTTTTAGA<br>
<strong>CTTGTGGAAAGGACGAAACACCG</strong>AGCAGCACAACAATATGGGTTTTAGAG<br>
<strong>CTTGTGGAAAGGACGAAACACCG</strong>CTCTTGGGTTTGGATGTTTGTTTTAGA<br>
<strong>CTTGTGGAAAGGACGAAACACCG</strong>TTTGGCGAGGGGAGCGCCGGTTTTAGA<br>
......</p>
<p>You can see that the first <strong>23</strong> nucleotides are identical, so in this case you need to tell MAGeCK to trim the first <strong>23</strong> nucleotides to collect read counts. <strong>If the nucleotide length in front of sgRNA varies between different reads, use <a class="" href="https://code.google.com/p/cutadapt/" rel="nofollow">cutadapt</a> to remove the adaptor sequences.</strong> <br>
</p>
<p>The sgRNA length can be determined from the experimental design. It is usually 20 nucleotide, but in this demo, the sgRNA length is <strong>19</strong>.</p>
<h3 id="step-4-run-the-mageck-count-command">Step 4: run the MAGeCK count command</h3>
<p>Now we have everything ready to generate count tables from MAGeCK. Place two fastq files and one library file into the same directory, and under that directory, run MAGeCK on terminal:</p>
<div class="codehilite"><pre>mageck count -l yusa_library.csv -n escneg --sample-label <span class="s2">"plasmid,ESC1"</span> --trim-5 <span class="m">23</span> --sgrna-len <span class="m">19</span> --fastq ERR376998.fastq  ERR376999.fastq
</pre></div>


<p>This command also tells MAGeCK to assign labels to each library ("plasmid" for ERR376998.fastq, and "ESC1" for ERR376999.fastq), trim the first 23 nucleotides, the sgRNA length is 19, and output the file with prefix "escneg".</p>
<p>If it is running successfully, you will see one file "escneg.count.txt" collecting all read counts. The top lines are as follows:</p>
<div class="codehilite"><pre>sgRNA   Gene    plasmid ESC1
chr19:5884430-5884453   SLC25A45        13      32
chr11:58831475-58831498 OLFR312 94      108
chr4:49282352-49282375  E130309F12RIK   85      128
</pre></div>


<p>If you use the <strong>--pdf-report</strong> option (see <a class="" href="http://sourceforge.net/p/mageck/wiki/visualization/">Visualization</a>), it will generate a nice PDF report of the sample statistics of the fastq files. Click <a class="" href="http://sourceforge.net/projects/mageck/files/pdf_sample/escneg_countsummary.pdf/download">Here</a> to see the PDF results.</p>
<h3 id="step-5-compare-samples-using-mageck-test-subcommand">Step 5: compare samples using MAGeCK test subcommand</h3>
<p>With the read count table, now you can compare ESC1 vs. plasmid condition to see which genes are negatively or positively selected:</p>
<div class="codehilite"><pre>mageck <span class="nb">test</span> -k escneg.count.txt -t ESC1 -c plasmid -n esccp
</pre></div>


<p>This command tells MAGeCK to compare ESC1 with plasmid in the read count table escneg.count.txt, and output results with prefix "esccp".</p>
<p>If successful, you should see a file "esccp.gene_summary.txt". The top lines are as follows:</p>
<div class="codehilite"><pre>id      num     neg|score  neg|p-value   neg|fdr neg|rank        neg|goodsgrna   pos|score  pos|p-value   pos|fdr pos|rank  pos|goodsgrna
GTF2B   5       2.0462e-10      2.5851e-07      0.000707        1       5       1.0     1.0     1.0     19150   0
RPS5    5       5.9353e-10      2.5851e-07      0.000707        2       5       1.0     1.0     1.0     19149   0
RPL19   4       2.695e-09       2.5851e-07      0.000707        3       4       1.0     1.0     1.0     19148   0
KIF18B  5       1.0136e-08      2.5851e-07      0.000707        4       5       1.0     1.0     1.0     19146   0
....
</pre></div>


<p>You can immediately see two ribosomal genes, RPS5 and RPL19, are on the top of negatively selected genes. If you rank the genes by "rank.pos" (11th column), you will see TRP53 (mouse homolog of TP53) on the top of positively selected genes:</p>
<div class="codehilite"><pre>sort -k 11,11n esccp.gene_summary.txt <span class="p">|</span> less

id      num     neg<span class="p">|</span>score  neg<span class="p">|</span>p-value   neg<span class="p">|</span>fdr neg<span class="p">|</span>rank        neg<span class="p">|</span>goodsgrna   pos<span class="p">|</span>score  pos<span class="p">|</span>p-value   pos<span class="p">|</span>fdr pos<span class="p">|</span>rank  pos<span class="p">|</span>goodsgrna
ZFP945  <span class="m">5</span>       1.0     1.0     0.999999        <span class="m">19150</span>   <span class="m">0</span>       9.6166e-07      5.4287e-06      0.05198 <span class="m">1</span>  5
TRP53   <span class="m">5</span>       0.95411 0.95409 0.999999        <span class="m">17901</span>   <span class="m">0</span>       1.0347e-06      5.4287e-06      0.05198 <span class="m">2</span>  4
PDAP1   <span class="m">5</span>       0.85937 0.86223 0.999999        <span class="m">15753</span>   <span class="m">1</span>       7.6412e-06      2.8178e-05      0.174505  <span class="m">3</span>       2
</pre></div>


<p>As is in the <strong>count</strong> command, if you use <strong>--pdf-report</strong> option, a nice PDF file will be generated. <a class="" href="http://sourceforge.net/projects/mageck/files/pdf_sample/esccp_summary.pdf/download">Here</a> is the example of generated PDF file in this demo.</p>
<h3 id="final-remarks">Final remarks</h3>
<p>Right now you should be quite familiar with basic functions of MAGeCK. MAGeCK also provides additional functions for you to further explore the data, for example, test the enrichment of pathways, plot the top-ranked genes or genes you are interested in, etc. If you have further questions, feel free to ask in our <a class="" href="https://groups.google.com/d/forum/mageck" rel="nofollow">google group</a>. Enjoy your MAGeCK trip!</p>
<h2 id="the-fourth-demo-using-mageck-mle-module">The fourth demo: using MAGeCK mle module</h2>
<p>Since version 0.5, MAGeCK provides a new subcommand, mle, to calculate gene essentiality from CRISPR screens. Compared with the original algorithm in "test" subcommand, MAGeCK-mle uses a measurement called <strong>beta score</strong> to call gene essentialities: a positive beta score means a gene is positively selected, and a negative beta score means a gene is negatively selected. It is similar to the term <strong>log fold change</strong> in differential expression, and compared with the original RRA algorithm, this measurement has the following advantages:</p>
<ul>
<li>It has only one score for one gene, instead of two scores in RRA: one for positive selection, one for negative selection;</li>
<li>It allows a direct comparison across multiple conditions, or even experiments;</li>
<li>It is able to incorporate sgRNA efficiency information.</li>
</ul>
<p>This demo will help you go through all the steps in running the mle module.</p>
<h3 id="step-1-download-the-count-table">Step 1: download the count table</h3>
<p>For simplicity, let's assume you already know how to generate read count table from fastq files; if not, check <a class="" href="https://sourceforge.net/p/mageck/wiki/demo/#the-third-demo-going-through-a-public-crisprcas9-screening-dataset">the third demo</a> above. We will use the read count table presented in <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336569" rel="nofollow">T Wang et al. Science 2014</a>.</p>
<p>Download the read count table <a class="" href="http://liulab.dfci.harvard.edu/Mageck/leukemia.csv.zip" rel="nofollow">here</a>, and unzip it to a csv file.</p>
<h3 id="step-2-prepare-the-design-matrix-file">Step 2: prepare the design matrix file</h3>
<p>The <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#design-matrix-file">design matrix</a> file indicates which sample is affected by which condition. It is generally a binary matrix indicating which sample (indicated by the first column) is affected by which condition (indicated by the first row). For the meanings of the design matrix, check the <a class="" href="https://sourceforge.net/p/mageck/wiki/input/">input file format</a> page. </p>
<p>To create a design matrix file, <strong>copy the following content to a text editing software, and save it as a plain txt file</strong>:</p>
<div class="codehilite"><pre>Samples        baseline        HL60        KBM7
HL60.initial   1               0           0
KBM7.initial   1               0           0
HL60.final     1               1           0
KBM7.final     1               0           1
</pre></div>


<p>Remember the following rules of a design matrix file:</p>
<ul>
<li>The design matrix file must include a header line of condition labels;</li>
<li>The first column is the sample labels that must match sample labels in read count file;</li>
<li>The second column must be a "baseline" column that sets all values to "1";</li>
<li>The element in the design matrix is either "0" or "1";</li>
<li>You must have at least one sample of "initial state" (e.g., day 0 or plasmid) that has only one "1" in the corresponding row. That only "1" must be in the baseline column.</li>
</ul>
<p>In the design matrix above, we have four samples, two corresponding to the initial states of two cell lines, and two corresponding to the final states of two cell lines. We design two conditions (HL60 and KBM7) that model the cell type-specific effects. </p>
<h3 id="step-3-run-the-module">Step 3: run the module</h3>
<p>Now we have the minimum requirements to run the MAGeCK mle module. Assuming you save the design matrix file as "designmat.txt", type the following command to run</p>
<div class="codehilite"><pre>mageck mle -k leukemia.new.csv -d designmat.txt -n beta_leukemia
</pre></div>


<p>If successful, MAGeCK mle will generate three files, the log file, the gene_summary file (including gene beta scores), and the sgrna_summary file (including sgRNA efficiency probability predictions). Here are a few lines of the gene_summary file:</p>
<div class="codehilite"><pre>Gene    sgRNA   HL60|beta       HL60|z  HL60|p-value    HL60|fdr        HL60|wald-p-value       HL60|wald-fdr   KBM7|beta       KBM7|z  KBM7|p-value    KBM7|fdr        KBM7|wald-p-value       KBM7|wald-fdr
RNF14   10      0.24927 0.72077 0.36256 0.75648 0.47105 0.9999  0.57276 1.6565  0.06468 0.32386 0.097625
0.73193
RNF10   10      0.10159 0.29373 0.92087 0.98235 0.76896 0.9999  0.11341 0.32794 0.90145 0.97365 0.74296 0.98421
RNF11   10      3.6354  10.513  0.0002811       0.021739        7.5197e-26      1.3376e-22      2.5928  7.4925  0.0014898       0.032024        6.7577e-14      1.33e-11
</pre></div>


<p>This file includes the beta scores in two conditions specified in the design matrix (HL60|beta and KBM7|beta), and the associated statistics. For more information, check the <a class="" href="https://sourceforge.net/p/mageck/wiki/output/#gene_summary_txt-in-mle-subcommand">output format specification of gene_summary file</a>. </p>
<h3 id="step-4-optional-including-the-sgrna-efficiency-into-mle-calculation">Step 4 (optional): including the sgRNA efficiency into mle calculation</h3>
<p>MAGeCK mle also allows an optional input of sgRNA efficiencies, which are calculated from <a class="" href="https://sourceforge.net/projects/spacerscoringcrispr">SSC</a>, a computational algorithm to predict sgRNA efficiencies from sgRNA sequence. To incorporate sgRNA efficiency information into the calculation, first download the SSC software, and compile the SSC program as indicated.</p>
<p>Next, we use the SSC program to calculate the sgRNA efficiencies. Download the library file related to this study <a class="" href="https://sourceforge.net/projects/mageck/files/libraries">from our collection of sgRNA libraries</a> (the file name is "tim.library.txt.zip"), and unzip it into a plain txt file. Here are a few lines of the library file:</p>
<div class="codehilite"><pre>Row.names       Sequence        Gene.Symbol
INO80B_m74682554        CATGGCCATGGGCACCCGCC    INO80B
NHS_p17705966   AGGAGCTGCACCGCCACGCC    NHS
MED14_m40594623 AACCACCAGCTGGTCCCGCC    MED14
</pre></div>


<p>THe library file format is slightly different from the input of SSC, so we need to treak it a little bit. SSC requires the first column of the input file to be sgRNA sequence, and the second column to be sgRNA label, so use the following command to adjust the column orders:</p>
<div class="codehilite"><pre>perl -ane <span class="s1">'print "$F[1]\t$F[0]\t$F[2]\n"'</span> tim.library.txt &gt; tim.library.in
</pre></div>


<p>The first few lines of "tim.library.in" are as follows:</p>
<div class="codehilite"><pre>Sequence        Row.names       Gene.Symbol
CATGGCCATGGGCACCCGCC    INO80B_m74682554        INO80B
AGGAGCTGCACCGCCACGCC    NHS_p17705966   NHS
AACCACCAGCTGGTCCCGCC    MED14_m40594623 MED14
</pre></div>


<p>Now we can use SSC to calculate sgRNA efficiencies:</p>
<div class="codehilite"><pre>./SSC -l <span class="m">20</span> -m SSC0.1/matrix/human_CRISPRi_20bp.matrix -i tim.library.in -o tim.library.out
</pre></div>


<p>where <em>-i</em> indicates the sgRNA sequence length, and <em>SSC0.1/matrix/human_CRISPRi_20bp.matrix</em> is the weight matrix provided in the SSC source. The few lines of "tim.library.out" are as follows:</p>
<div class="codehilite"><pre>CATGGCCATGGGCACCCGCC    INO80B_m74682554        INO80B  -0.005239
AGGAGCTGCACCGCCACGCC    NHS_p17705966   NHS     -0.018779
AACCACCAGCTGGTCCCGCC    MED14_m40594623 MED14   -0.018132
</pre></div>


<p>The last column (4th column) is the sgRNA efficiency score calculated by SSC.</p>
<p>Finally, use this file as input of the MAGeCK mle module:</p>
<div class="codehilite"><pre>mageck mle -k leukemia.new.csv -d designmat.txt -n beta_leukemia --sgrna-efficiency tim.library.ko.out --sgrna-eff-name-column <span class="m">1</span> --sgrna-eff-score-column 3
</pre></div>


<p>Here, three optional arguments related to sgRNA efficiencies are provided. <em>--sgrna-efficiency</em> provides the file name of the efficiency prediction, <em>--sgrna-eff-name-column</em> provides the column index corresponding to the sgRNA names (index = 1 means the name column is the 2nd column), and the <em>--sgrna-eff-score-column</em> provides the column index corresponding to the sgRNA scores (index=3, or the 4th column).</p>
<p>This command will tell MAGeCK mle to incorporate sgRNA efficiency information into the calculation.</p>
<p>For more information of the SSC program, refer to the following publication:</p>
<p><a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/26063738" rel="nofollow">Xu et al. Genome Research 2015 Aug;25(8):1147-57</a></p>
<p>Return to <a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p>
<hr></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="usage">Usage</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#usage">Usage</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#run">run</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#test">test</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#count">count</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway">pathway</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#mle">mle</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#plot">plot</a></li>
</ul>
</li>
</ul>
</div>
<p>The main portal of MAGeCK is the <strong>mageck</strong> program, which includes a couple of different subprograms:</p>
<ul>
<li>run: collect sgRNA read counts from read mapping files (sam format), and perform sgRNA and gene ranking.</li>
<li>count: only collect sgRNA read counts from read mapping files (sam format).</li>
<li>test: given a table of read counts, perform the sgRNA and gene ranking.</li>
<li>pathway: given a ranked gene list, test whether one pathway is enriched.</li>
<li>mle: perform maximum-likelihood estimation of gene essentiality scores.</li>
</ul>
<p>There is also another subprogram <em>plot</em> that plots some figures of the genes you are interested in from the <em>test</em> results.</p>
<ul>
<li>plot: Generating graphics for selected genes.</li>
</ul>
<h2 id="run">run</h2>
<p>This subcommand allows you generate comparison results directly from fastq files, with limited parameter settings available. The parameters for the run sub-command are included in <em>test</em> and <em>count</em> sub-command. See both sub-commands for more details. <strong>It is strongly suggested that users run the count and test command separately, in order to gain a finer control of the results.</strong></p>
<h2 id="test">test</h2>
<p>This subcommand tests and ranks sgRNAs and genes based on the read count tables provided.</p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">test</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">-</span><span class="n">k</span> <span class="n">COUNT_TABLE</span> <span class="o">-</span><span class="n">t</span> <span class="n">TREATMENT_ID</span> <span class="o">[-</span><span class="n">c</span> <span class="n">CONTROL_ID</span><span class="o">]</span>
               <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span> <span class="o">[--</span><span class="n">norm</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">none</span><span class="o">,</span><span class="n">median</span><span class="o">,</span><span class="n">total</span><span class="o">}]</span>
               <span class="o">[--</span><span class="n">normcounts</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">file</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">gene</span><span class="o">-</span><span class="n">test</span><span class="o">-</span><span class="n">fdr</span><span class="o">-</span><span class="n">threshold</span> <span class="n">GENE_TEST_FDR_THRESHOLD</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">adjust</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">fdr</span><span class="o">,</span><span class="n">holm</span><span class="o">}]</span> <span class="o">[--</span><span class="n">variance</span><span class="o">-</span><span class="n">from</span><span class="o">-</span><span class="n">all</span><span class="o">-</span><span class="n">samples</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">sort</span><span class="o">-</span><span class="n">criteria</span> <span class="o">{</span><span class="n">neg</span><span class="o">,</span><span class="n">pos</span><span class="o">}]</span> <span class="o">[--</span><span class="n">keep</span><span class="o">-</span><span class="n">tmp</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">control</span><span class="o">-</span><span class="n">sgrna</span> <span class="n">CONTROL_SGRNA</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">pdf</span><span class="o">-</span><span class="n">report</span><span class="o">]</span>
</pre></div>


<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-k COUNT_TABLE, --count-table COUNT_TABLE</td>
<td>Provide a tab-separated count table instead of sam files. Each line in the table should include sgRNA name (1st column), targeting gene (2nd column) and read counts in each sample. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-read-count-file">input/#sgrna-read-count-file</a> for a detailed description.</td>
</tr>
<tr>
<td>-t TREATMENT_ID, --treatment-id TREATMENT_ID</td>
<td>Sample label or sample index (0 as the first sample) in the count table as treatment experiments, separated by comma (,). If sample label is provided, the labels must match the labels in the first line of the count table; for example, "HL60.final,KBM7.final". For sample index, "0,2" means the 1st and 3rd samples are treatment experiments. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input/#sample-index</a> for a detailed description.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>-c CONTROL_ID, --control-id CONTROL_ID</td>
<td>Sample label or sample index in the count table as control experiments, separated by comma (,). Default is all the samples not specified in treatment experiments. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input/#sample-index</a> for a detailed description.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--norm-method {none,median,total}</td>
<td>Method for normalization, default median.</td>
</tr>
<tr>
<td>--normcounts-to-file</td>
<td>Write normalized read counts to file ({output-prefix}.normalized.txt).</td>
</tr>
<tr>
<td>--gene-test-fdr-threshold GENE_TEST_FDR_THRESHOLD</td>
<td>FDR threshold for gene test, default 0.25.</td>
</tr>
<tr>
<td>--adjust-method {fdr,holm,pounds}</td>
<td>Method for sgrna-level p-value adjustment, including false discovery rate (fdr), holm's method (holm), or  pounds's method (pounds).</td>
</tr>
<tr>
<td>--variance-from-all-samples</td>
<td>Estimate the variance from all samples, instead of from only control samples. Use this option only if you believe there are relatively few essential sgRNAs or genes between control and treatment samples.</td>
</tr>
<tr>
<td>--sort-criteria {neg,pos}</td>
<td>Sorting criteria, either by negative selection (neg) or positive selection (pos). Default negative selection.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
<tr>
<td>--control-sgrna CONTROL_SGRNA</td>
<td>A list of control sgRNAs for generating proper null distribution.</td>
</tr>
<tr>
<td>--pdf-report</td>
<td>Generate pdf report of the analysis.</td>
</tr>
</tbody>
</table>
<h2 id="count">count</h2>
<p>This subcommand collects sgRNA read count information from fastq files. The output count tables can be used directly in the <em>test</em> subcommand.</p>
<p><em>usage</em>: </p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">count</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">[-</span><span class="n">l</span> <span class="n">LIST_SEQ</span><span class="o">]</span> <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
                <span class="o">[--</span><span class="n">sample</span><span class="o">-</span><span class="n">label</span> <span class="n">SAMPLE_LABEL</span><span class="o">]</span> <span class="o">[--</span><span class="n">trim</span><span class="o">-</span><span class="mi">5</span> <span class="n">TRIM_5</span><span class="o">]</span>
                <span class="o">[--</span><span class="n">sgrna</span><span class="o">-</span><span class="n">len</span> <span class="n">SGRNA_LEN</span><span class="o">]</span> <span class="o">[--</span><span class="n">count</span><span class="o">-</span><span class="n">n</span><span class="o">]</span> <span class="o">--</span><span class="n">fastq</span> <span class="n">FASTQ</span> <span class="o">[</span><span class="n">FASTQ</span> <span class="o">...]</span>
                <span class="o">[--</span><span class="n">pdf</span><span class="o">-</span><span class="n">report</span><span class="o">]</span> <span class="o">[--</span><span class="n">keep</span><span class="o">-</span><span class="n">tmp</span><span class="o">]</span>
</pre></div>


<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>--fastq</td>
<td>Sample fastq files, separated by space; use comma (,)  to indicate technical replicates of the same sample. For example, "--fastq sample1_replicate1.fastq,sample1_replicate2.fastq sample2_replicate1.fastq,sample2_replicate2.fastq" indicates two samples with 2 technical replicates for each sample.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>-l LIST_SEQ, --list-seq LIST_SEQ</td>
<td>A file containing list of sgRNA names, the sequences and target genes, either in .txt or in .csv format. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">input/#sgrna-library-file</a> for more details. If this file is not provided, mageck will count all possible sgRNAs in the fastq.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--sample-label SAMPLE_LABEL</td>
<td>Sample labels, separated by comma (,). Must be equal to the number of samples provided (in --fastq option). Default "sample1,sample2,...".</td>
</tr>
<tr>
<td>--trim-5 TRIM_5</td>
<td>Length of trimming the 5' of the reads. Default 0</td>
</tr>
<tr>
<td>--sgrna-len SGRNA_LEN</td>
<td>Length of the sgRNA. Default 20</td>
</tr>
<tr>
<td>--count-n</td>
<td>Count sgRNAs with Ns. By default, sgRNAs containing Ns will be discarded.</td>
</tr>
<tr>
<td>--pdf-report</td>
<td>Generate pdf report of the fastq files.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
<tr>
<td>--unmapped-to-file</td>
<td>Save unmapped reads to file.</td>
</tr>
</tbody>
</table>
<h2 id="pathway">pathway</h2>
<p>MAGeCK can also invoke RRA to test if a pathway is enriched in one particular gene ranking. </p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">pathway</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">--</span><span class="n">gene</span><span class="o">-</span><span class="n">ranking</span> <span class="n">GENE_RANKING</span> <span class="o">--</span><span class="n">gmt</span><span class="o">-</span><span class="n">file</span> <span class="n">GMT_FILE</span>
                  <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
                  <span class="o">[--</span><span class="n">sort</span><span class="o">-</span><span class="n">criteria</span> <span class="o">{</span><span class="n">neg</span><span class="o">,</span><span class="n">pos</span><span class="o">}]</span> <span class="o">[--</span><span class="n">keep</span><span class="o">-</span><span class="n">tmp</span><span class="o">]</span>
                  <span class="o">[--</span><span class="n">ranking</span><span class="o">-</span><span class="n">column</span> <span class="n">RANKING_COLUMN</span><span class="o">]</span>
                  <span class="o">[--</span><span class="n">ranking</span><span class="o">-</span><span class="n">column</span><span class="o">-</span><span class="mi">2</span> <span class="n">RANKING_COLUMN_2</span><span class="o">]</span>
</pre></div>


<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>--gene-ranking GENE_RANKING</td>
<td>The gene ranking file generated by the gene test step.</td>
</tr>
<tr>
<td>--gmt-file GMT_FILE</td>
<td>The pathway file in GMT format. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#pathway-file-gmt">input/#pathway-file-gmt</a> for more details of the GMT file format.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>--single-ranking</td>
<td>The provided file is a (single) gene ranking file, either positive or negative selection. Only one enrichment comparison will be performed.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--sort-criteria {neg,pos}</td>
<td>Sorting criteria, either by negative selection (neg) or positive selection (pos). Default negative selection.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
<tr>
<td>--ranking-column RANKING_COLUMN</td>
<td>Column number or label in gene summary file for gene ranking; can be either an integer of column number, or a string of column label. Default "2" (the 3rd column).</td>
</tr>
<tr>
<td>--ranking-column-2 RANKING_COLUMN_2</td>
<td>Column number or label in gene summary file for gene ranking; can be either an integer of column number, or a string of column label. This option is used to determine the column for positive selections and is disabled if --single-ranking is specified. Default "5" (the 6th column).</td>
</tr>
</tbody>
</table>
<h2 id="mle">mle</h2>
<p>The mle subcommand performs maximum-likelihood analysis of gene essentialities, instead of the RRA analysis.</p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">mle</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">norm</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">none</span><span class="o">,</span><span class="n">median</span><span class="o">,</span><span class="n">total</span><span class="o">,</span><span class="n">control</span><span class="o">}]</span>
                   <span class="o">[--</span><span class="n">control</span><span class="o">-</span><span class="n">sgrna</span> <span class="n">CONTROL_SGRNA</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">genes</span><span class="o">-</span><span class="n">varmodeling</span> <span class="n">GENES_VARMODELING</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">permutation</span><span class="o">-</span><span class="n">round</span> <span class="n">PERMUTATION_ROUND</span><span class="o">]</span>
                   <span class="o">[-</span><span class="n">i</span> <span class="n">INCLUDE_SAMPLES</span><span class="o">]</span> <span class="o">[-</span><span class="n">b</span> <span class="n">BETA_LABELS</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">remove</span><span class="o">-</span><span class="n">outliers</span><span class="o">]</span> <span class="o">[--</span><span class="n">adjust</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">fdr</span><span class="o">,</span><span class="n">holm</span><span class="o">,</span><span class="n">pounds</span><span class="o">}]</span>
                   <span class="o">[--</span><span class="n">sgrna</span><span class="o">-</span><span class="n">efficiency</span> <span class="n">SGRNA_EFFICIENCY</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">sgrna</span><span class="o">-</span><span class="n">eff</span><span class="o">-</span><span class="n">name</span><span class="o">-</span><span class="n">column</span> <span class="n">SGRNA_EFF_NAME_COLUMN</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">sgrna</span><span class="o">-</span><span class="n">eff</span><span class="o">-</span><span class="n">score</span><span class="o">-</span><span class="n">column</span> <span class="n">SGRNA_EFF_SCORE_COLUMN</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">update</span><span class="o">-</span><span class="n">efficiency</span><span class="o">]</span> <span class="o">-</span><span class="n">k</span> <span class="n">COUNT_TABLE</span> <span class="o">-</span><span class="n">d</span> <span class="n">DESIGN_MATRIX</span>
</pre></div>


<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-k COUNT_TABLE, --count-table COUNT_TABLE</td>
<td>Provide a tab-separated count table. Each line in the  table should include sgRNA name (1st column), target gene (2nd column) and read counts in each sample. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-read-count-file">input/#sgrna-read-count-file</a> for a detailed description.</td>
</tr>
<tr>
<td>-d DESIGN_MATRIX, --design-matrix DESIGN_MATRIX</td>
<td>Provide a design matrix, either a file name or a quoted string of the design matrix. For example,  "1,1;1,0". The row of the design matrix must match the  order of the samples in the count table (if --include-samples is not specified), or the order of the samples  by the --include-samples option.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--norm-method {none,median,total,control}</td>
<td>Method for normalization, including "none" (no normalization), "median" (median normalization, default), "total" (normalization by total read  counts), "control" (normalization by control sgRNAs  specified by the --control-sgrna option).</td>
</tr>
<tr>
<td>--control-sgrna CONTROL_SGRNA</td>
<td>A list of control sgRNAs.</td>
</tr>
<tr>
<td>--genes-varmodeling GENES_VARMODELING</td>
<td>The number of genes for mean-variance modeling. Default 1000.</td>
</tr>
<tr>
<td>--permutation-round PERMUTATION_ROUND</td>
<td>The rounds for permutation (interger). The permutation  time is (# genes)*x for x rounds of permutation. Suggested value: 100 (may take longer time). Default 10.</td>
</tr>
<tr>
<td>-i INCLUDE_SAMPLES, --include-samples INCLUDE_SAMPLES</td>
<td>Specify the sample labels if the design matrix is not  given by file in the --design-matrix option. Sample labels are separated by ",", and must match the labels in the count table.</td>
</tr>
<tr>
<td>-b BETA_LABELS, --beta-labels BETA_LABELS</td>
<td>Specify the labels of the variables (i.e., beta), if the design matrix is not given by file in the  --design-matrix option. Should be separated by ",", and the number of labels must equal to (# columns of  design matrix), including baseline labels. Default value: "bata_0,beta_1,beta_2,...".</td>
</tr>
<tr>
<td>--remove-outliers</td>
<td>Try to remove outliers. Turning this option on will slow the algorithm.</td>
</tr>
<tr>
<td>--adjust-method {fdr,holm,pounds}</td>
<td>Method for sgrna-level p-value adjustment, including  false discovery rate (fdr), holm's method (holm), or pounds's method (pounds).</td>
</tr>
</tbody>
</table>
<h2 id="plot">plot</h2>
<p>The plot command generating graphics for selected genes. <strong>For interactive visualizations, use our new <a class="" href="https://bitbucket.org/liulab/mageck-vispr/overview" rel="nofollow">MAGeCK-VISPR</a> algorithm.</strong></p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">plot</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">-</span><span class="n">k</span> <span class="n">COUNT_TABLE</span> <span class="o">-</span><span class="n">g</span> <span class="n">GENE_SUMMARY</span> <span class="o">[--</span><span class="n">genes</span> <span class="n">GENES</span><span class="o">]</span>
                   <span class="o">[-</span><span class="n">s</span> <span class="n">SAMPLES</span><span class="o">]</span> <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
                   <span class="o">[--</span><span class="n">norm</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">none</span><span class="o">,</span><span class="n">median</span><span class="o">,</span><span class="n">total</span><span class="o">}]</span> <span class="o">[--</span><span class="n">keep</span><span class="o">-</span><span class="n">tmp</span><span class="o">]</span>
</pre></div>


<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-k COUNT_TABLE, --count-table COUNT_TABLE</td>
<td>Provide a tab-separated count table.</td>
</tr>
<tr>
<td>-g GENE_SUMMARY, --gene-summary GENE_SUMMARY</td>
<td>The gene summary file generated by the test command.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>--genes GENES</td>
<td>A list of genes to be plotted, separated by comma. Default: none.</td>
</tr>
<tr>
<td>-s SAMPLES, --samples SAMPLES</td>
<td>A list of samples to be plotted, separated by comma. Default: using all samples in the count table.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--norm-method {none,median,total}</td>
<td>Method for normalization, default median.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
</tbody>
</table>
<p>Return to <a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p>
<hr></div>
  
</div>


<div>
  
    <div class="markdown_content"><div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#visualization-functions-in-mageck">Visualization Functions in MAGeCK</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-pdf-report-option">The --pdf-report option</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-plot-command">The plot command</a></li>
</ul>
</li>
</ul>
</div>
<h1 id="visualization-functions-in-mageck">Visualization Functions in MAGeCK</h1>
<p>Since version 0.5, MAGeCK enables a couple of visualization functions. With these features on, MAGeCK helps users better interpret datasets and results, and generates figures and tables that can be directly used in presentations or papers.</p>
<p>An example of images generated by <em>plot</em> command:</p>
<p><img height="400" rel="nofollow" src="./MAGeCK_Wiki_Home_files/sample1_summary.png" width="400"></p>
<p>Here is another example of images generated by <em>count</em> command:</p>
<p><img height="400" rel="nofollow" src="./MAGeCK_Wiki_Home_files/abl_smalllib_stat.png" width="400"></p>
<p>The Visualization function has additional software dependencies, but they are easy to install in many operating systems. See  <a class="" href="https://sourceforge.net/p/mageck/wiki/install">installation</a> for more details.</p>
<h2 id="the-pdf-report-option">The --pdf-report option</h2>
<p>MAGeCK will generate PDF files in both <em>count</em> and <em>test</em> command, by simply adding the <em>--pdf-report</em> option. If successful, a &lt;prefix&gt;.countsummary.pdf (for <em>count</em> command) or &lt;prefix&gt;_summary.pdf (for <em>test</em> command) will be generated. </p>
<p>You can also try it in the two demos provided. In demo1, note that the command used in the <em>run.sh</em> is:</p>
<div class="codehilite"><pre>mageck test -k sample.txt -t HL60.final,KBM7.final -c HL60.initial,KBM7.initial  -n demo
</pre></div>


<p>Use the following command to generate PDF file:</p>
<div class="codehilite"><pre>mageck test -k sample.txt -t HL60.final,KBM7.final -c HL60.initial,KBM7.initial  -n demo --pdf-report
</pre></div>


<p>You can download the sample PDF file from demo1 <a class="" href="http://sourceforge.net/projects/mageck/files/pdf_sample/demo_summary.pdf/download">here</a>.</p>
<p>In demo2, the command used in the <em>runmageck.sh</em> is:</p>
<div class="codehilite"><pre>mageck run --fastq test1.fastq test2.fastq -l library.txt -n demo --sample-label L1,CTRL -t L1 -c CTRL
</pre></div>


<p>You can split the <em>run</em> command into <em>count</em> and <em>test</em> command, with <em>--pdf-report</em> option enabled. An alternative way is, note that the Rnw and R files (used for PDF file production) exist after successfully running this demo:</p>
<div class="codehilite"><pre>demo.count.median_normalized.csv  demo.count.txt         demo.R                  library.txt   test2.fastq
demo_countsummary.R               demo.gene_summary.txt  demo.sgrna_summary.txt  runmageck.sh
demo_countsummary.Rnw             demo.log               demo_summary.Rnw        test1.fastq
</pre></div>


<p>Simply execute the two .R files and you can get the PDF files as well:</p>
<div class="codehilite"><pre>Rscript demo_countsummary.R
Rscript demo.R
</pre></div>


<p>You can download the count sample PDF file from demo2 <a class="" href="http://sourceforge.net/projects/mageck/files/pdf_sample/demo_countsummary.pdf/download">here</a>.</p>
<h2 id="the-plot-command">The plot command</h2>
<p>After running test, MAGeCK can generate a couple of figures describing the genes you are interested using the <em>plot</em> command. In the demo1 for example, if you are interested in the ACTR8 gene, use the following command to generate the PDF reports describing the sgRNA read count change of ACTR8, and its RRA score relative to all-gene RRA score distribution:<br>
</p>
<div class="codehilite"><pre>mageck plot -k sample.txt -g demo.gene_summary.txt --genes ACTR8
</pre></div>


<p>The PDF file generated using this command is <a class="" href="http://sourceforge.net/projects/mageck/files/pdf_sample/sample1_summary.pdf/download">here</a>.</p>
<p><a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="input-file-specification">Input file specification</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#input-file-specification">Input file specification</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna-read-count-file">sgRNA read count file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sample-index">Sample index</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#design-matrix-file">design matrix file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna-library-file">sgRNA library file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway-file-gmt">pathway file (gmt)</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrnagene-mapping-file-depreciated-after-version-03">sgRNA/gene mapping file (depreciated after version 0.3)</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="sgrna-read-count-file">sgRNA read count file</h2>
<p>The sgRNA read count file will be used in <em>-k</em> parameter in the <em>test</em> or <em>run</em> sub-command.</p>
<p>The read count file should list the names of the sgRNA, the gene it is targeting, followed by the read counts in each sample. Each item should be separated by the tab ('\t'). A header line is optional. For example in the studies of <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336569" rel="nofollow">T. Wang et al. Science 2014</a>, there are 4 CRISPR screening samples, and they are labeled as: HL60.initial, KBM7.initial, HL60.final, KBM7.final. Here are a few lines of the read count file: </p>
<div class="codehilite"><pre>sgRNA           gene    HL60.initial    KBM7.initial    HL60.final      KBM7.final
A1CF_m52595977  A1CF    213             274            883                175
A1CF_m52596017  A1CF    294             412            1554              1891
A1CF_m52596056  A1CF    421             368            566                759
A1CF_m52603842  A1CF    274             243            314                855
A1CF_m52603847  A1CF    0               50             145                266
</pre></div>


<p>The <em>count</em> sub-command will output the read count file like this.</p>
<h2 id="sample-index">Sample index</h2>
<p>In the <em>-t/--treatment-id</em>, <em>-c/--control-id</em> parameters, you can use either sample label or sample index to specify samples. If sample label is used, the labels <a class="alink notfound" href="https://sourceforge.net/p/mageck/wiki/must">[must]</a> match the sample labels in the first line of the count table. For example, "HL60.final,KBM7.final".</p>
<p>You can also use sample index to specify samples. The index of the sample is the order it appears in the sgRNA read count file, starting from 0. The index is used in the <em>-t/--treatment-id</em>, <em>-c/--control-id</em> parameters. In the example above, there are four samples, and the index of each sample is as follows:</p>
<table>
<thead>
<tr>
<th>sample</th>
<th>index</th>
</tr>
</thead>
<tbody>
<tr>
<td>HL60.initial</td>
<td>0</td>
</tr>
<tr>
<td>KBM7.initial</td>
<td>1</td>
</tr>
<tr>
<td>HL60.final</td>
<td>2</td>
</tr>
<tr>
<td>KBM7.final</td>
<td>3</td>
</tr>
</tbody>
</table>
<h2 id="design-matrix-file">design matrix file</h2>
<p>The design matrix is a txt file indicating the effects of different conditions on different samples. In this file, each row is a sample, each column is a condition, and the value is 1 or 0, indicating whether the sample (in the row) is affected by the condition (in the column).</p>
<p>Here is a simple example of the design matrix from the studies in  <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336569" rel="nofollow">T. Wang et al. Science 2014</a>. The CRISPR screens are done on two cell lines, HL60 and KBM7, and four samples are generated, two corresponding to the initial states of two cell lines, and two corresponding to the final states of two cell lines. If you want to model the effects of two cell lines, you can have the design matrix as follows:</p>
<div class="codehilite"><pre>Samples        baseline        HL60        KBM7
HL60.initial   1               0           0
KBM7.initial   1               0           0
HL60.final     1               1           0
KBM7.final     1               0           1
</pre></div>


<p>Here are some important rules of the design matrix:</p>
<ul>
<li>The design matrix file must include a header line of condition labels;</li>
<li>The first column is the sample labels that must match labels in read count file (see the above example in sgRNA read count file);</li>
<li>The second column must be a "baseline" column that sets all values to "1";</li>
<li>The element in the design matrix is either "0" or "1".</li>
<li>You must have at least one sample of "initial state" (e.g., day 0 or plasmid) that has only one "1" in the corresponding row. That only "1" must be in the baseline column.</li>
</ul>
<h2 id="sgrna-library-file">sgRNA library file</h2>
<p>When starting from fastq files, MAGeCK needs to know the sgRNA sequence and its targeting gene. Such information is provided in the sgRNA library file, and can be specified by the <em>-l/--list-seq</em> option in <strong>run</strong> or <strong>count</strong> subcommand.</p>
<p><strong>The sgRNA library file can be provided either in .txt format or in .csv format.</strong> There are three columns in the library file: the sgRNA ID, the sequence, and the gene it is targeting. One example of the library file is provided as <em>library.txt</em> in demo2:</p>
<div class="codehilite"><pre>s_10007 TGTTCACAGTATAGTTTGCC    CCNA1
s_10008 TTCTCCCTAATTGCTTGCTG    CCNA1
s_10027 ACATGTTGCTTCCCCTTGCA    CCNC
</pre></div>


<p>If provided in .csv format, the file will look like:</p>
<div class="codehilite"><pre>s_10007,TGTTCACAGTATAGTTTGCC,CCNA1
s_10008,TTCTCCCTAATTGCTTGCTG,CCNA1
s_10027,ACATGTTGCTTCCCCTTGCA,CCNC
</pre></div>


<h2 id="pathway-file-gmt">pathway file (gmt)</h2>
<p>The GMT file format stores the pathway information and is consistent with the GMT file in Gene Set Enrichment Analysis (GSEA). The details of the GMT format can be found at <a class="" href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" rel="nofollow">GSEA website</a>.</p>
<p>You can also download different pathway files directly from GSEA <a class="" href="http://www.broadinstitute.org/gsea/downloads.jsp" rel="nofollow">MSigDB</a> database. They can be used directly by MAGeCK. </p>
<h2 id="sgrnagene-mapping-file-depreciated-after-version-03">sgRNA/gene mapping file (depreciated after version 0.3)</h2>
<p>The sgRNA/gene mapping file will be used in the <em>--gene-test</em> parameter in the <em>test</em> or <em>run</em> sub-command.</p>
<p>This file should list the names of the sgRNAs and their corresponding genes, separated by the tab ('\t'). For example:</p>
<div class="codehilite"><pre>A1CF_m52595977  A1CF
A1CF_m52596017  A1CF
A1CF_m52596056  A1CF
A1CF_m52603842  A1CF
A1CF_m52603847  A1CF
A1CF_p52595870  A1CF
A1CF_p52595881  A1CF
A1CF_p52596023  A1CF
</pre></div>


<p>Return to <a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p>
<hr></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="output-file-specification">Output file specification</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#output-file-specification">Output file specification</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna_summary_txt">sgrna_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#gene_summary_txt">gene_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#gene_summary_txt-in-mle-subcommand">gene_summary_txt in mle subcommand</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway_summary_txt">pathway_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#log">log</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#rnw-and-r">Rnw and R</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#intermediate-file-formats">Intermediate file formats</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">gene_txt</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<p>The output of the MAGeCK consists of the following files:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna_summary_txt">sgrna_summary.txt</a>: The sgRNA ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_summary_txt">gene_summary.txt</a>: The gene ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#pathway_summary_txt">pathway_summary.txt</a>: The pathway ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#log">log</a>: The logging information during the running.</li>
</ul>
<p>Other file formats are intermediate files, including:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">.gene.high.txt</a>:  The gene ranking results (positively selected genes).</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">.gene.low.txt</a>:   The gene ranking results (negatively selected genes).</li>
</ul>
<h2 id="sgrna_summary_txt">sgrna_summary_txt</h2>
<p>An example of the sgRNA ranking results is as follows:</p>
<div class="codehilite"><pre>sgrna   Gene   control_count   treatment_count control_mean    treat_mean      control_var     adj_var score   p.low   p.high  p.twosided      FDR     high_in_treatment
INO80B_m74682554   INO80B        0.0/0.0 1220.1598778/1476.14096301      0.810860655738  1348.15042041   0.0     19.0767988005   308.478081895   1.0     1.11022302463e-16       2.22044604925e-16       1.57651669497e-14       True
NHS_p17705966   NHS   1.62172131148/3.90887850467     2327.09368635/1849.95115143     2.76529990807   2088.52241889   2.6155440132    68.2450168229   252.480744404   1.0     1.11022302463e-16       2.22044604925e-16       1.57651669497e-14       True
</pre></div>


<p>The contents of each column are as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>sgrna</td>
<td>sgRNA ID</td>
</tr>
<tr>
<td>Gene</td>
<td>The targeting gene</td>
</tr>
<tr>
<td>control_count</td>
<td>Normalized read counts in control samples</td>
</tr>
<tr>
<td>treatment_count</td>
<td>Normalized read counts in treatment samples</td>
</tr>
<tr>
<td>control_mean</td>
<td>Mean read counts in control samples</td>
</tr>
<tr>
<td>treat_mean</td>
<td>Mean read counts in treatment samples</td>
</tr>
<tr>
<td>control_var</td>
<td>The raw variance in control samples</td>
</tr>
<tr>
<td>adj_var</td>
<td>The adjusted variance in control samples</td>
</tr>
<tr>
<td>score</td>
<td>The score of this sgRNA</td>
</tr>
<tr>
<td>p.low</td>
<td>p-value (lower tail)</td>
</tr>
<tr>
<td>p.high</td>
<td>p-value (higher tail)</td>
</tr>
<tr>
<td>p.twosided</td>
<td>p-value (two sided)</td>
</tr>
<tr>
<td>FDR</td>
<td>false discovery rate</td>
</tr>
<tr>
<td>high_in_treatment</td>
<td>Whether the abundance is higher in treatment samples</td>
</tr>
</tbody>
</table>
<h2 id="gene_summary_txt">gene_summary_txt</h2>
<p>An example of the gene summary file is as follows:</p>
<div class="codehilite"><pre>id      num     neg|score  neg|p-value   neg|fdr neg|rank        neg|goodsgrna   pos|score  pos|p-value   pos|fdr pos|rank  pos|goodsgrna
ESPL1   12      6.4327e-10      7.558e-06       7.9e-05 1       11      0.99725 0.99981 0.999992        615     0
RPL18   12      6.4671e-10      7.558e-06       7.9e-05 2       11      0.99799 0.99989 0.999992        620     0
CDK1    12      2.6439e-09      7.558e-06       7.9e-05 3       12      1.0     0.99999 0.999992        655     0
</pre></div>


<p>The contents of each column is as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>id</td>
<td>Gene ID</td>
</tr>
<tr>
<td>num</td>
<td>The number of targeting sgRNAs for each gene</td>
</tr>
<tr>
<td>neg|score</td>
<td>The RRA lo value of this gene in negative selection</td>
</tr>
<tr>
<td>neg|p-value</td>
<td>The raw p-value (using permutation) of this gene in negative selection</td>
</tr>
<tr>
<td>neg|fdr</td>
<td>The false discovery rate of this gene in negative selection</td>
</tr>
<tr>
<td>neg|rank</td>
<td>The ranking of this gene in negative selection</td>
</tr>
<tr>
<td>neg|goodsgrna</td>
<td>The number of "good" sgRNAs, i.e., sgRNAs whose ranking is below the alpha cutoff (determined by the <a class="" href="http://sourceforge.net/p/mageck/wiki/usage/#test">--gene-test-fdr-threshold</a> option), in negative selection.</td>
</tr>
<tr>
<td>pos|score</td>
<td>The number of targeting sgRNAs for each gene in positive selection (usually the same as num.neg)</td>
</tr>
<tr>
<td>pos|score</td>
<td>The RRA lo value of this gene in negative selection</td>
</tr>
<tr>
<td>pos|p-value</td>
<td>The raw p-value of this gene in positive selection</td>
</tr>
<tr>
<td>pos|fdr</td>
<td>The false discovery rate of this gene in positive selection</td>
</tr>
<tr>
<td>pos|rank</td>
<td>The ranking of this gene in positive selection</td>
</tr>
<tr>
<td>pos|goodsgrna</td>
<td>The number of "good" sgRNAs, i.e., sgRNAs whose ranking is below the alpha cutoff (determined by the <a class="" href="http://sourceforge.net/p/mageck/wiki/usage/#test">--gene-test-fdr-threshold</a> option), in positive selection.</td>
</tr>
</tbody>
</table>
<p>Genes are ranked by the p.neg field (by default). If you need a ranking by the p.pos, you can use the --sort-criteria option.</p>
<h2 id="gene_summary_txt-in-mle-subcommand">gene_summary_txt in mle subcommand</h2>
<p>The output of the gene_summary.txt in mle subcommand is pretty similar to the gene_summary.txt format above, except a few new columns. Here is an example of the gene_summary.txt generated from the mle subcommand:</p>
<div class="codehilite"><pre>Gene    sgRNA   HL60|beta       HL60|z  HL60|p-value    HL60|fdr        HL60|wald-p-value       HL60|wald-fdr   KBM7|beta       KBM7|z  KBM7|p-value    KBM7|fdr        KBM7|wald-p-value       KBM7|wald-fdr
RNF14   10      0.24927 0.72077 0.36256 0.75648 0.47105 0.9999  0.57276 1.6565  0.06468 0.32386 0.097625
    0.73193
RNF10   10      0.10159 0.29373 0.92087 0.98235 0.76896 0.9999  0.11341 0.32794 0.90145 0.97365 0.74296 0.98421
RNF11   10      3.6354  10.513  0.0002811       0.021739        7.5197e-26      1.3376e-22      2.5928  7.4925  0.0014898       0.032024        6.7577e-14      1.33e-11
</pre></div>


<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>Gene</td>
<td>Gene ID</td>
</tr>
<tr>
<td>sgRNA</td>
<td>The number of targeting sgRNAs for each gene</td>
</tr>
<tr>
<td>HL60|beta;KBM7|beta</td>
<td>The beta scores of this gene in conditions "HL60" and "KBM7", respectively. The conditions are specified in the design matrix as an input of the mle subcommand.</td>
</tr>
<tr>
<td>HL60|p-value</td>
<td>The raw p-value (using permutation) of this gene</td>
</tr>
<tr>
<td>HL60|fdr</td>
<td>The false discovery rate of this gene</td>
</tr>
<tr>
<td>HL60|wald-p-value</td>
<td>The p value using Wald test</td>
</tr>
<tr>
<td>HL60|wald-fdr</td>
<td>The false discovery rate of the Wald test</td>
</tr>
</tbody>
</table>
<h2 id="pathway_summary_txt">pathway_summary_txt</h2>
<p>The output of the pathway summary is similar to the gene summary. Here is an example:</p>
<div class="codehilite"><pre>id      num     neg|score  neg|p-value   neg|fdr neg|rank        neg|goodsgrna   pos|score  pos|p-value   pos|fdr pos|rank  pos|goodsgrna
KEGG_RIBOSOME   87      8.3272e-23      2.6473e-05      0.001238        1       50      0.051213        0.20927 0.841006        38      4
KEGG_SPLICEOSOME        125     3.7084e-08      2.6473e-05      0.001238        2       41      0.52219 0.80968 0.99902 149     13
KEGG_PROTEASOME 44      1.9586e-06      2.6473e-05      0.001238        3       18      0.52149 0.80905 0.99902 148     5
</pre></div>


<p>This table shows a pathway <em>KEGG_RIBOSOME</em> has 87 genes, its RRA lo value 8.3272e-23, permutation p value is 2.6473e-05 (negative selection), FDR 0.001238, its ranking is 1, and there are 50 genes that are below the alpha cutoff. This shows the genes in this pathway (i.e., ribosomal genes) are strongly negatively selected, which is expected in negative selection CRISPR experiments.</p>
<h2 id="log">log</h2>
<p>This file includes the logging information during the execution. For <em>count</em> command, it will list some basic statistics of the dataset at the end, including the number of reads, the number of reads mapped to the library, the number of zero-count sgRNAs, etc.</p>
<h2 id="rnw-and-r">Rnw and R</h2>
<p>If the "--pdf-report" option is on for <em>count</em> or <em>test</em> command, MAGeCK may generate Rnw and R files that are used to create PDF files. MAGeCK calls the <a class="" href="http://users.stat.umn.edu/~geyer/Sweave/" rel="nofollow">Sweave</a> function in R to generate PDF files.</p>
<h2 id="intermediate-file-formats">Intermediate file formats</h2>
<p>These files will be automatically deleted after the completion of each command. To keep these files, use the "--keep-tmp" option during the execution.</p>
<h3 id="gene_txt">gene_txt</h3>
<p>An example of the gene ranking file (.gene.high.txt or .gene.low.txt) is as follows:</p>
<div class="codehilite"><pre> group_id        #_items_in_group        lo_value        FDR
 RPL3    93      4.9169e-36      0.000080
 RPL8    67      1.8232e-24      0.000080
 RPS2    61      1.6928e-20      0.000080
 RPS18   40      1.0152e-18      0.000080
</pre></div>


<p>The contents of each column is as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>group_id</td>
<td>Gene ID</td>
</tr>
<tr>
<td>#_items_in_group</td>
<td>The number of targeting sgRNAs for each gene</td>
</tr>
<tr>
<td>lo_value</td>
<td>The raw p-value</td>
</tr>
<tr>
<td>FDR</td>
<td>The false discovery rate</td>
</tr>
</tbody>
</table>
<p>Return to <a class="alink" href="./MAGeCK_Wiki_Home_files/MAGeCK_Wiki_Home.html">[Home]</a></p>
<hr></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="commonly-used-libraries">Commonly used libraries</h1>
<p><a class="" href="https://sourceforge.net/projects/mageck/files/libraries/">Download frequently used libraries</a></p>
<p>For your convenience, we provide a set of library files that are ready to be used in MAGeCK (in the -l/--list-seq option of the count command) in the <a class="" href="https://sourceforge.net/projects/mageck/files/libraries/">libraries</a> folder. You can also create your own library files, see <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">sgrna-library-file</a> for more details.</p>
<table>
<thead>
<tr>
<th>File</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>Human_GeCKOv2_Library_A_3_mageck.csv.zip</td>
<td>Human <a class="" href="http://www.addgene.org/crispr/libraries/geckov2/" rel="nofollow">GeCKO v2</a> half-library A (can be used in either 1- or 2-plasmid systems)</td>
</tr>
<tr>
<td>Human_GeCKOv2_Library_B_1_mageck.csv.zip</td>
<td>Human GeCKO v2 half-library B</td>
</tr>
<tr>
<td>Human_GeCKOv2_Library_combine.csv.zip</td>
<td>Human GeCKO v2 combined library of A and B</td>
</tr>
<tr>
<td>mouse_geckov2_library_a_2_mageck.csv.zip</td>
<td>Mouse GeCKO v2 half-library A (can be used in either 1- or 2-plasmid systems)</td>
</tr>
<tr>
<td>mouse_geckov2_library_b_1_mageck.csv.zip</td>
<td>Mouse GeCKO v2 half-library B</td>
</tr>
<tr>
<td>mouse_geckov2_library_combine.csv.zip</td>
<td>Mouse GeCKO v2 combined library of A and B</td>
</tr>
<tr>
<td>GeCKOv1.txt.zip</td>
<td>GeCKO v1 library file (from the GeCKO <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336571" rel="nofollow">Science paper</a>)</td>
</tr>
<tr>
<td>human_sam_library.csv.zip</td>
<td>Human <a class="" href="http://www.addgene.org/crispr/libraries/sam/" rel="nofollow">Synergistic Activation Mediator (SAM)</a> pooled library (CRISPRa library), generated by Feng Zhang laboratory.</td>
</tr>
<tr>
<td>yusa_library.csv.zip</td>
<td>Mouse knockout library generated by <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24535568" rel="nofollow">Kosuke Yusa laboratory</a>.</td>
</tr>
<tr>
<td>tim_library.txt.zip</td>
<td>Human CRISPR knockout library of 7,000 genes (from <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336569" rel="nofollow">T. Waing Science 2014</a>).</td>
</tr>
</tbody>
</table></div>
  
</div>


<div>
  
    <div class="markdown_content"><h1 id="q-and-a">Q and A</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#q-and-a">Q and A</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation-problems">Installation problems</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#i-encountered-an-error-after-installation-importerror-no-module-named-mageck-what-is-the-problem">I encountered an error after installation: "ImportError: No module named mageck". What is the problem?</a></li>
</ul>
</li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#counting-sgrnas-from-fastq-files">Counting sgRNAs from fastq files</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#how-to-deal-with-biological-replicates-and-technical-replicates">How to deal with biological replicates and technical replicates?</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-trim-5-option-can-only-trim-a-fixed-length-of-nucleotides-before-sgrna-but-what-if-the-trimming-length-is-different-in-different-reads">The --trim-5 option can only trim a fixed length of nucleotides before sgRNA, but what if the trimming length is different in different reads?</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-get-the-simple-statistics-of-the-fastq-files">How do I get the simple statistics of the fastq files?</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-know-the-quality-of-my-samples">How do I know the quality of my samples?</a></li>
</ul>
</li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#interpreting-results">Interpreting results</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-know-if-my-experiments-work-well">How do I know if my experiments work well?</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#i-see-very-few-genes-that-are-below-the-certain-fdr-cutoff-like-010-why-it-is-that-and-what-should-i-do">I see very few genes that are below the certain FDR cutoff (like 0.10). Why it is that and what should I do?</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#what-does-the-control-sgrna-control_sgrna-option-do-how-to-use-this-option">What does the --control-sgrna CONTROL_SGRNA option do? How to use this option?</a></li>
</ul>
</li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#visualization">Visualization</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-test-or-count-command-is-successful-but-i-have-some-problems-producing-the-pdf-file-how-can-i-generate-the-pdf-file">The test or count command is successful but I have some problems producing the PDF file. How can I generate the PDF file?</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<p>You can always ask questions on our <a class="" href="https://groups.google.com/forum/?hl=en#!forum/mageck" rel="nofollow">Google group</a>. Usually your questions are also other's questions, so please help us better improve our algorithm by joining our Google group and asking questions there!</p>
<h2 id="installation-problems">Installation problems</h2>
<h3 id="i-encountered-an-error-after-installation-importerror-no-module-named-mageck-what-is-the-problem">I encountered an error after installation: "ImportError: No module named mageck". What is the problem?</h3>
<p><strong>A</strong>: Probably you are installing MAGeCK to your own directory, which is not recognized by Python. The solution is to set up the PYTHONPATH environment: see <a class="" href="https://sourceforge.net/p/mageck/wiki/install/#setting-up-the-environment-variables">install/#setting-up-the-environment-variables</a> for more details.</p>
<h2 id="counting-sgrnas-from-fastq-files">Counting sgRNAs from fastq files</h2>
<h3 id="how-to-deal-with-biological-replicates-and-technical-replicates">How to deal with biological replicates and technical replicates?</h3>
<p><strong>A</strong>: Usually you can pool the read counts for technical replicates of the same sample. To do this, use comma (,) to separate the fastq files of the technical replicates from the same sample in the --fastq option. For example, "--fastq sample1_replicate1.fastq,sample1_replicate2.fastq sample2_replicate1.fastq,sample2_replicate2.fastq" indicates two samples with 2 technical replicates for each sample.<br>
For biological replicates, treat them as separate samples and use them together when doing the comparison; so MAGeCK can analyze the variance of these samples. For example in the test command, "-t sample1_bio_replicate1,sample1_bio_replicate2 -c sample2_bio_replicate1,sample2_bio_replicate2" compares 2 samples (with 2 biological replicates in each sample).</p>
<h3 id="the-trim-5-option-can-only-trim-a-fixed-length-of-nucleotides-before-sgrna-but-what-if-the-trimming-length-is-different-in-different-reads">The --trim-5 option can only trim a fixed length of nucleotides before sgRNA, but what if the trimming length is different in different reads?</h3>
<p><strong>A</strong>: You can use <a class="" href="https://code.google.com/p/cutadapt/" rel="nofollow">cutadapt</a> to trim the adaptor sequences of variable length before running MAGeCK.</p>
<h3 id="how-do-i-get-the-simple-statistics-of-the-fastq-files">How do I get the simple statistics of the fastq files?</h3>
<p><strong>A</strong>: Since version 0.5, MAGeCK produces a "countsummary.txt" file containing all the statistics of the fastq files. If you use "--pdf-report" option, the statistics of fastq files are also in the PDF file from the test. </p>
<p>The statistics can also be found in the log file (for run and count command). Here is an example of the log file generated from count command (the last few lines):</p>
<div class="codehilite"><pre>INFO  @ Mon, 02 Feb 2015 08:12:15: Summary of file sample1_R1.fastq: 
INFO  @ Mon, 02 Feb 2015 08:12:15: reads        45631055 
INFO  @ Mon, 02 Feb 2015 08:12:15: mappedreads  34300176 
INFO  @ Mon, 02 Feb 2015 08:12:15: zerosgrnas   119315 
INFO  @ Mon, 02 Feb 2015 08:12:15: label        sample_1 
INFO  @ Mon, 02 Feb 2015 08:12:15: Summary of file sample2_R1.fastq: 
INFO  @ Mon, 02 Feb 2015 08:12:15: reads        36344414 
INFO  @ Mon, 02 Feb 2015 08:12:15: mappedreads  27042629 
INFO  @ Mon, 02 Feb 2015 08:12:15: zerosgrnas   119002 
INFO  @ Mon, 02 Feb 2015 08:12:15: label        sample_2
</pre></div>


<p>It provides the total number of reads, the number of mapped reads, the number of sgRNAs with 0 read counts, and the sample label of the fastq file.</p>
<h3 id="how-do-i-know-the-quality-of-my-samples">How do I know the quality of my samples?</h3>
<p><strong>A</strong>: You can take a look at the <a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-get-the-simple-statistics-of-the-fastq-files">sample statistics</a>. Generally in a good negative selection sample, (1) the mapped reads should be over 60 percent of the total number reads, and (2) the number of zero-count sgRNAs should be few (&lt;5%, and prefered &lt;1%). One exception is in positive selection experiments, where the number of zero-count sgRNAs may be much higher, but the percentage of mapped reads should be reasonably high. </p>
<p>You can also inspect the results by taking a look at the comparison results, see the <a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-know-if-my-experiments-work-well">related question</a> below.</p>
<h2 id="interpreting-results">Interpreting results</h2>
<h3 id="how-do-i-know-if-my-experiments-work-well">How do I know if my experiments work well?</h3>
<p><strong>A</strong>: First of all, make sure your sample statistics looks good (see the <a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#how-do-i-know-the-quality-of-my-samples">related question</a> in "Counting sgRNAs from fastq files"). Next, take a look at the rankings of some well-known genes. In negative selection experiments, you will expect some ribosomal genes and well-known oncogenes that are on the top; for example, MYC, RAS, etc. In positive selection experiments, TP53 usually has a high ranking.<br>
</p>
<p>Besides visually inspecting top-ranked genes, a good validation is to run the <strong>pathway</strong> command to test on <em>MSigDB KEGG pathways</em> (<a class="" href="http://www.broadinstitute.org/gsea/msigdb/collections.jsp" rel="nofollow">see MSigDB website</a>). In negative selection experiments (usually on some condition compared with day 0 condition), you will expect to see a set of essential pathways ranking on the top, like ribosome, splicesome, proteasome and cell cycle genes. If you see these pathways coming out, this is a good sign that your experiments are working. The smaller the RRA lo_value and p values they have, the better they are.</p>
<h3 id="i-see-very-few-genes-that-are-below-the-certain-fdr-cutoff-like-010-why-it-is-that-and-what-should-i-do">I see very few genes that are below the certain FDR cutoff (like 0.10). Why it is that and what should I do?</h3>
<p><strong>A</strong>: There are a couple of reasons that the top ranked genes have a high FDR. First, many CRISPR/Cas9 libraries designed few sgRNAs (&lt;7) for each gene. Since some of them may have low cutting efficiency or off-target effect, there may not be enough statistical power to detect essential genes. Second, if there are two many comparisons (or genes), the multiple comparison adjustment may lead to a high FDR estimation. Also, MAGeCK employs a pretty stringent statistical framework to evaluate the statistical significance, its FDR estimation may be conservative.<br>
</p>
<p>There are a couple of procedures you can do to increase the sensitivity. First, try to filter out genes that you think are not hits before running MAGeCK; for example, remove genes that have extremely low expression, genes that have very few targeting sgRNAs (&lt;4). Second, If you have a list of negative control genes (genes that you think are not essential, like AAVS1), you can specify the corresponding sgRNA IDs using the <strong>--control-sgrna</strong> option, thus allowing MAGeCK to have a better estimation of null distribution.</p>
<h3 id="what-does-the-control-sgrna-control_sgrna-option-do-how-to-use-this-option">What does the <strong>--control-sgrna CONTROL_SGRNA</strong> option do? How to use this option?</h3>
<p><strong>A</strong>: This option is used to generate null distribution when calculating the p values. If this option is not specified, MAGeCK generates the null distribution of RRA scores by assuming all of the genes in the library are non-essential. This approach is sometimes over-conservative, and you can improve this if you know some genes are not essential. By providing the corresponding sgRNA IDs in the <strong>--control-sgrna</strong> option, MAGeCK will have a better estimation of p values. </p>
<p>To use this option, you need to prepare a text file specifying the IDs of control sgRNAs, one line for one sgRNA ID.<br>
</p>
<h2 id="visualization">Visualization</h2>
<h3 id="the-test-or-count-command-is-successful-but-i-have-some-problems-producing-the-pdf-file-how-can-i-generate-the-pdf-file">The test or count command is successful but I have some problems producing the PDF file. How can I generate the PDF file?</h3>
<p><strong>A</strong>: MAGeCK will generate <em>.R and </em>.Rnw file even if the "--pdf-report" option is not specified. You can copy these files to a new computer where both R and pdflatex are properly installed, and use the following command to generate PDF files:</p>
<div class="codehilite"><pre>Rscript *.R
</pre></div>


<p>Note the for <em>count</em> command, the median-normalized read count file (<em>.median_normalized.csv) should also be copied to the same directory. For </em>test<em> command, the gene summary file (</em>.gene_summary.txt) should also be copied to the same directory.</p></div>
  
</div>


<h6>Project Members:</h6>
	<ul class="md-users-list">
		<li><a href="https://sourceforge.net/u/davidliwei/">Wei Li</a> (admin)</li>
		
	</ul>
</div></div></div>

  <div id="create_wiki_page_holder" title="Create New Page" style="display:none">
    <form>
        <label class="grid-2">Name</label>
        <div class="grid-7"><input type="text" name="name" tabindex="3"></div>
    </form>
  </div>

                </div>
                
                
  <hr style="margin-top: 1em; margin-bottom: 2em; clear:both;">
  
  
    <h4>Related</h4>
    <p>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/QA/">Wiki: QA</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/demo/">Wiki: demo</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/history/">Wiki: history</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/input/">Wiki: input</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/install/">Wiki: install</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/libraries/">Wiki: libraries</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/output/">Wiki: output</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/usage/">Wiki: usage</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/visualization/">Wiki: visualization</a><br>
    
    </p>
  
  
      
          <div id="discussion_holder"><h2>Discussion</h2>
          
          <div>

<div>
                <div class="row">
                        <div class="column grid_12">
      
        <div>
  <div class="page_list">
    
  </div>
  <div class="clear"></div>
</div>
      
      <div id="comment">
        
          
        
          <div id="allow_moderate"></div>
        
      </div>
      <div style="clear:both"></div>
      
      <div style="clear:both"></div>
    </div>
  </div>
</div>

  <div id="new_post_holder">
    <span id="add_comment"></span>
    <div class="row reply">
      <div class="grid-2 tcenter">
          <p class="gravatar">
            
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5(1)" alt="Wei Li" title="Wei Li" class="emboss x48">
  <br>
            <small>
               
        
            <a href="https://sourceforge.net/u/davidliwei/">
                
                    Wei Li
                
            </a>
        
    
            </small>
          </p>
        </div>
        <div class="grid-17">
        
<div>
  <form id="edit_post" method="post" action="https://sourceforge.net/p/mageck/wiki/_discuss/thread/8575cbc1/post" enctype="multipart/form-data">
    
    
<div class="markdown_edit">
  <textarea id="w-12f.text" name="Xhw2ZCxccWULAtdOdv84h97g567s" class="None" tabindex="4" style="display: none;"></textarea><div class="editor-toolbar"><a title="Bold (⌘+B)" class="fa fa-bold"></a><a title="Italic (⌘+I)" class="fa fa-italic"></a><i class="separator">|</i><a title="Quote (⌘+&#39;)" class="fa fa-quote-left"></a><a title="Generic List (⌘+L)" class="fa fa-list-ul"></a><a title="Numbered List (⌘+⌥+L)" class="fa fa-list-ol"></a><i class="separator">|</i><a title="Create Link (⌘+K)" class="fa fa-link"></a><a title="Insert Image (⌘+⌥+I)" class="fa fa-picture-o"></a><i class="separator">|</i><a class="fa fa-eye"></a><a class="fa fa-question-circle"></a></div><div class="CodeMirror cm-s-paper CodeMirror-wrap"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 14.900390625px; left: 14.9005737304688px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" style="position: absolute; padding: 0px; width: 1000px; height: 1em; outline: none;" tabindex="4"></textarea></div><div class="CodeMirror-vscrollbar" cm-not-content="true" style="min-width: 18px;"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-hscrollbar" cm-not-content="true" style="min-height: 18px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 30px; min-height: 25px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><span><span>​</span>x</span></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-cursors"><div class="CodeMirror-cursor" style="left: 3.99148559570313px; top: 0px; height: 17.2734375px;">&nbsp;</div></div><div class="CodeMirror-code"><pre class=" CodeMirror-line "><span style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 25px;"></div><div class="CodeMirror-gutters" style="display: none; height: 90px;"></div></div></div><div class="editor-statusbar"><span class="autosave"></span><span class="lines">0</span><span class="words">0</span><span class="cursor">0:0</span></div>
  <div class="modal markdown_help" style="display:none">
    <b data-icon="D" class="ico ico-close close"></b>
    <div class="markdown_help_contents" data-url="/p/mageck/wiki/markdown_syntax_dialog"></div>
  </div>
  <input type="hidden" class="markdown_project" value="mageck" tabindex="5">
  <input type="hidden" class="markdown_neighborhood" value="4be2faf8898e33156f00003d" tabindex="6">
  <input type="hidden" class="markdown_app" value="wiki" tabindex="7">
</div>
    <input type="hidden" class="original_value" value="" tabindex="8">
    <span class="arw"><span></span></span><br>
    <div style="clear:both"></div>
    <input type="submit" value="Post" tabindex="9">
    <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="btn link cancel_edit_post">Cancel</a> &nbsp;
    <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="btn link attachment_form_add_button">Add attachments</a>
    <input type="file" class="text attachment_form_fields" style="display:none" multiple="True" name="file_info" tabindex="10">
  
  <input id="w-130" name="timestamp" type="hidden" value="1439134100" tabindex="11">
  <input id="w-131" name="spinner" type="hidden" value="Xg3n8c2PoW9OdMR1iLQFnodpgAhc" tabindex="12">
  <p class="XnyGbQAQOOHNXkpbYmUqsJTkY">
    <label for="XhBGTHQaRa-NpM4w_qc-YMxUmVHU">You seem to have CSS turned off.
        Please don't fill out this field.</label><br>
    <input id="XhBGTHQaRa-NpM4w_qc-YMxUmVHU" name="XhRGTHQaRayefoEDm4_71bpw2YE4" type="text" tabindex="13"><br></p>
  <p class="XnyGbQAQOOHNXkpbYmUqsJTkY">
    <label for="XhBGTHQaRa-JpM4w_qc-YMxUmVHU">You seem to have CSS turned off.
        Please don't fill out this field.</label><br>
    <input id="XhBGTHQaRa-JpM4w_qc-YMxUmVHU" name="XhRGTHQaRaiefoEDm4_71bpw2YE4" type="text" tabindex="14"><br></p>
    
    <input name="_session_id" type="hidden" value="c846f8a43f86bfeb92e51efd1e450acdaba3d04723d963fce266b322fa4a94bd9146e5b0afb4d0c4" tabindex="15">
  
  </form>
</div>
      </div>
    </div>
  </div>
</div>
          </div>
      
  
  <div id="lightbox_confirm" class="modal" style="display:none">
  <b data-icon="D" class="ico ico-close close"></b>
  
    
  
</div>
  
    <form action="https://sourceforge.net/p/mageck/wiki/subscribe" id="wiki_subscribe_form" style="display:none;" method="POST">
      
    <input name="_session_id" type="hidden" value="c846f8a43f86bfeb92e51efd1e450acdaba3d04723d963fce266b322fa4a94bd9146e5b0afb4d0c4" tabindex="16">
  
      
        <input type="hidden" name="subscribe" value="1" tabindex="17">
      
    </form>
  

            </div>
        
    </div>
</section>
  
<footer id="site-footer">
    <div class="wrapper">
        <nav>
            <h5>SourceForge</h5>
            <a href="https://sourceforge.net/about">About</a>
            <a href="https://sourceforge.net/blog/category/sitestatus/">Site Status</a>
            <a href="http://twitter.com/sfnet_ops">@sfnet_ops</a>
            <a id="allura-notice" href="http://allura.apache.org/">
                <p>Powered by</p>
                <p>Apache Allura™</p>
                <img src="./MAGeCK_Wiki_Home_files/logo-black-svg_g.png">
            </a>
        </nav>
        <nav>
            <h5>Find and Develop Software</h5>
            <a href="https://sourceforge.net/create/">Create a Project</a>
            <a href="https://sourceforge.net/directory/">Software Directory</a>
            <a href="https://sourceforge.net/top">Top Downloaded Projects</a>
        </nav>
        <nav>
            <h5>Community</h5>
            <a href="https://sourceforge.net/blog/">Blog</a>
            <a href="http://twitter.com/sourceforge">@sourceforge</a>
            <a href="https://sourceforge.net/jobs?source=footer">Job Board</a>
            <a href="http://library.slashdotmedia.com/?source=sfnet_footer">Resources</a>
        </nav>
        <nav>
            <h5>Help</h5>
            <a href="http://p.sf.net/sourceforge/docs">Site Documentation</a>
            <a href="https://sourceforge.net/support">Support Request</a>
            <a href="http://p.sf.net/sourceforge/irc">Real-Time Support</a>
        </nav>
    </div>
</footer>
<footer id="site-copyright-footer">
    <div class="wrapper">
        <div id="copyright">
            © 2015 Slashdot Media. All Rights Reserved.<br>
            <div id="dhi-icon"><span class="logo-DHI-alt"></span></div>
            <div id="service-text"><div class="smalltext"> SourceForge is a <a href="http://www.dhigroupinc.com/" target="_blank">DHI service</a></div></div>
        </div>
        <nav>
            <a href="http://slashdotmedia.com/terms-of-use">Terms</a>
            <a href="http://slashdotmedia.com/privacy-statement/">Privacy</a>
            <span id="teconsent"></span>
            <a href="http://slashdotmedia.com/opt-out-choices">Opt Out Choices</a>
            <a href="http://slashdotmedia.com/">Advertise</a>
        </nav>
    </div>
</footer>
<div id="messages" style="position: fixed; top: 20px;">
    
</div>


    <!-- ew:body_js -->


    <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/js(2)"></script>

    
<!-- /ew:body_js -->



    <!-- ew:body_js_tail -->


    <script type="text/javascript">$(function () {
            var $lightbox = $('#lightbox_confirm');
            var $trigger = $('a.post-link');
            $trigger.bind('click', function(e) {
                $lightbox.lightbox_me({ modalCSS: { minHeight: 0, width: 'inherit', top: '150px'}});
                return false;
            });
            $($lightbox).delegate('.close', 'click', function(e) {
                $lightbox.trigger('close');
                return false;
            });
        });
$(function () {
            var $lightbox = $('#lightbox_create_wiki_page');
            var $trigger = $('#sidebar a.add_wiki_page');
            $trigger.bind('click', function(e) {
                $lightbox.lightbox_me();
                return false;
            });
            $($lightbox).delegate('.close', 'click', function(e) {
                $lightbox.trigger('close');
                return false;
            });
        });
$(function () {
            $('#lightbox_create_wiki_page form').submit(function(){
                location.href = $('#sidebar a.add_wiki_page').attr('href') +
                    encodeURIComponent($('input[name=name]', $(this)).val().replace('/', '-')) + '/edit';
                return false;
            });
        });
$(document).ready(function () {
            $("a.attachment_form_add_button").click(function(evt){
                $(this).hide();
                $(".attachment_form_fields", this.parentNode).show();
                evt.preventDefault();
            });
            $("a.cancel_edit_post").click(function(evt){
                $("textarea", this.parentNode).val($("input.original_value", this.parentNode).val());
                $(".attachment_form_fields input", this.parentNode).val('');
                evt.preventDefault();
            });
         });

        (function () {
            $('div.discussion-post').each(function () {
                var post = this;
                $('.submit', post).button();
                $('.flag_post', post).click(function (ele) {
                    this.parentNode.submit();
                    return false;
                });
                $('.moderate_post', post).click(function(e){
                    e.preventDefault();
                    var mod = $(this).text();
                    if (mod === 'Delete' && !confirm('Really delete this post?')) {
                        return;
                    }
                    var id_post = $(post).attr('id');
                    $.ajax({
                        type: 'POST',
                        url: this.parentNode.action,
                        data: jQuery(this.parentNode).serialize(),
                        success: function() {
                            if (mod == 'Delete'){
                                $(post).remove();
                            }
                            else if (mod == 'Approve'){
                                $('a.reply_post, a.shortlink, form.moderate_spam, form.moderate_approve', post).toggle();
                                $('div.moderate', post).removeClass('moderate');
                            }
                            else if (mod == 'Spam'){
                                $(post).remove();
                            }
                        }
                    });
                });

                function get_cm($elem) { return $('.CodeMirror', $elem)[0].CodeMirror; }

                if($('a.edit_post', post)){
                    $('a.edit_post', post).click(function (ele) {
                        $('.display_post', post).hide();
                        var $edit_post_form = $('.edit_post_form', post);
                        var cm = get_cm($edit_post_form);
                        $edit_post_form.show();
                        cm.refresh();
                        cm.focus();
                        return false;
                    });
                    $("a.cancel_edit_post", post).click(function(evt){
                        $('.display_post', post).show();
                        $('.edit_post_form', post).hide();
                    });
                }
                if($('.reply_post', post)){
                    $('.reply_post', post).click(function (ele) {
                        var $reply_post_form = $('.reply_post_form', post);
                        var cm = get_cm($reply_post_form);
                        $reply_post_form.show();
                        cm.focus();
                        return false;
                    });
                    $('.reply_post', post).button();
                }
                if($('.add_attachment', post)){
                    $('.add_attachment', post).click(function (ele) {
                        $('.add_attachment_form', post).show();
                        return false;
                    });
                }
                if($('.shortlink', post)){
                    var popup = $('.shortlink_popup', post);
                    $('.shortlink', post).click(function(evt){
                        evt.preventDefault();
                        popup.lightbox_me({
                            onLoad: function() {
                                $('input', popup).select();
                            }
                        });
                    });
                    $('.close', popup).bind('click', function() {
                        popup.hide();
                    });
                }
            });
        }());
        
$(function () {
            $('select.results_per_page').change(function () {
                this.form.submit();});});
$(function () {
          $('input.label_edit').tagsInput({
              'height':'100%',
              'width':'100%',
              'autocomplete_url':'/p/mageck/wiki/tags'
          });
          $('form').on('blur', '.ui-autocomplete-input', function() {
              setTimeout(function(){
                  var clicked = $(document.activeElement); // This is the element that has focus
                  if (clicked.is('#ui-active-menuitem')) {
                      return false;
                  } else {
                      var value = $('div.tagsinput div input').val();
                      var exists = $('input.label_edit').tagExist(value);
                      var default_value = $('div.tagsinput div input').attr('data-default');
                      if ((value !== default_value) && (!exists) && value !== '') {
                          $('input.label_edit').addTag(value);
                      }
                      $('input[type=submit]', this).prop('disabled', true);
                  }
                }, 1);
            });
        });

        $(document).ready(function () {
            var thread_tag = $('a.thread_tag');
            var thread_spam = $('a.sidebar_thread_spam');
            var tag_thread_holder = $('#tag_thread_holder');
            var allow_moderate = $('#allow_moderate');
            var mod_thread_link = $('#mod_thread_link');
            var mod_thread_form = $('#mod_thread_form');
            if (mod_thread_link.length) {
                if (mod_thread_form.length) {
                    mod_thread_link.click(function (e) {
                        mod_thread_form.show();
                        return false;
                    });
                }
            }
            if (thread_tag.length) {
                if (tag_thread_holder.length) {
                    var submit_button = $('input[type="submit"]', tag_thread_holder);
                    var cancel_button = $('<a href="#" class="btn link">Cancel</a>').click(function(evt){
                        evt.preventDefault();
                        tag_thread_holder.hide();
                        thread_tag.removeClass('active');
                    });
                    submit_button.after(cancel_button);
                    thread_tag.click(function (e) {
                        tag_thread_holder.show();
                        thread_tag.addClass('active');
                        // focus the submit to scroll to the form, then focus the subject for them to start typing
                        submit_button.focus();
                        $('input[type="text"]', tag_thread_holder).focus();
                        return false;
                    });
                }
            }
            if (thread_spam.length) {
                if (allow_moderate.length) {
                    var cval = $.cookie('_session_id');
                    thread_spam[0].style.display='block';
                }
            }
        });
        

            $(function(){$(".download-button-533ec35981b24b2aced2594a").load("/projects/mageck/best_release.html");
        });</script>

    
<!-- /ew:body_js_tail -->



  <div id="lightbox_create_wiki_page" class="modal" style="display:none">
  <b data-icon="D" class="ico ico-close close"></b>
  
    
    <h1>Add a Wiki Page</h1>
    <form class="grid-10">
        <label class="grid-2">Name</label>
        <div class="grid-7"><input type="text" name="name" tabindex="18"></div>
        <label class="grid-2">&nbsp;</label>
        <div class="grid-7"><input type="submit" value="Create page" tabindex="19"></div>
    </form>
  
  
</div>
    <script type="text/javascript">
        $('.post-link').click(function () {
            var version = $(this).data("dialog-id");
            if (version) {
                var dialog_text = $(".confirmation_dialog_" + version);
            } else {
                var dialog_text = $(".confirmation_dialog");
            }

            var modal = $('#lightbox_confirm');
            modal.html(dialog_text.html());

            modal.find('.continue_confirm').click(function (evt) {
                var cval = $.cookie('_session_id');
                evt.preventDefault();
                $.post($(this).attr('href'), {_session_id: cval}, function (val) {
                    window.location = val.location;
                }, 'json');
            });
        });
    </script>
  
  <script type="text/javascript">
    $('a[href$="#toggle-subscribe"]').click(function() {
      $('#wiki_subscribe_form').submit();
      return false;
    })
  </script>



    


    <!-- Google Code for Remarketing tag -->
    <!-- Remarketing tags may not be associated with personally identifiable information or placed on pages related to sensitive categories. For instructions on adding this tag and more information on the above requirements, read the setup guide: google.com/ads/remarketingsetup -->
    <script type="text/javascript">
        /* <![CDATA[ */
        var google_conversion_id = 1002083962;
        var google_conversion_label = "G_uGCOaBlAQQ-qzq3QM";
        var google_custom_params = window.google_tag_params;
        var google_remarketing_only = true;
        /* ]]> */
    </script>
    <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/conversion.js"> </script><iframe name="google_conversion_frame" title="Google conversion frame" width="0" height="0" src="about:blank" frameborder="0" marginwidth="0" marginheight="0" vspace="0" hspace="0" allowtransparency="true" scrolling="no" style="display: none !important; visibility: hidden !important; opacity: 0 !important; background-position: 300px 13px;">&lt;img height="1" width="1" border="0" alt="" src="https://googleads.g.doubleclick.net/pagead/viewthroughconversion/1002083962/?frame=0&amp;random=1439134103589&amp;cv=7&amp;fst=1439134103589&amp;num=1&amp;fmt=1&amp;label=G_uGCOaBlAQQ-qzq3QM&amp;guid=ON&amp;u_h=900&amp;u_w=1440&amp;u_ah=812&amp;u_aw=1440&amp;u_cd=24&amp;u_his=50&amp;u_tz=-240&amp;u_java=true&amp;u_nplug=6&amp;u_nmime=8&amp;frm=0&amp;url=https%3A//sourceforge.net/p/mageck/wiki/Home/&amp;ref=https%3A//sourceforge.net/p/mageck/wiki/demo/" /&gt;</iframe>
    <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/get" id="truste_0.8830538387410343"></script><script src="./MAGeCK_Wiki_Home_files/notice" async="async" crossorigin=""></script>
    <noscript>
      &lt;div style="display:inline;"&gt;
        &lt;img height="1" width="1" style="border-style:none;" alt="" src="//googleads.g.doubleclick.net/pagead/viewthroughconversion/1002083962/?value=0&amp;amp;label=G_uGCOaBlAQQ-qzq3QM&amp;amp;guid=ON&amp;amp;script=0"/&gt;
      &lt;/div&gt;
    </noscript>

     
      

<script>
    $(document).ready(function () {
        $(".tooltip").tooltipster({
            animation: 'fade',
            delay: 200,
            theme: 'tooltipster-light',
            trigger: 'hover',
            position: 'right',
            iconCloning: false,
            maxWidth: 300
        }).focus(function () {
            $(this).tooltipster('show');
        }).blur(function () {
            $(this).tooltipster('hide');
        });
    })
</script>

<script id="hiddenlpsubmitdiv" style="display: none;"></script><script>try{for(var lastpass_iter=0; lastpass_iter < document.forms.length; lastpass_iter++){ var lastpass_f = document.forms[lastpass_iter]; if(typeof(lastpass_f.lpsubmitorig2)=="undefined"){ lastpass_f.lpsubmitorig2 = lastpass_f.submit; if (typeof(lastpass_f.lpsubmitorig2)=='object'){ continue;}lastpass_f.submit = function(){ var form=this; var customEvent = document.createEvent("Event"); customEvent.initEvent("lpCustomEvent", true, true); var d = document.getElementById("hiddenlpsubmitdiv"); if (d) {for(var i = 0; i < document.forms.length; i++){ if(document.forms[i]==form){ if (typeof(d.innerText) != 'undefined') { d.innerText=i.toString(); } else { d.textContent=i.toString(); } } } d.dispatchEvent(customEvent); }form.lpsubmitorig2(); } } }}catch(e){}</script></body></html>